---
title: "GQ_AF_markdown.R"
author: "heloise_calzan"
date: "2025-02-13"
output: pdf_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```


```{r cars}
setwd("E:\\github\\BILL_2025\\R")


# Step 1: Load necessary libraries
library(tidyverse)
library(gridExtra)
library(dplyr)
```

# Analyse préliminaires Genotype quality & Allelic frequency

```{r cars}
# Step 2: Import the data
af_data <- read.table("AF_per_SV.txt", header = FALSE, col.names = c("Frequency", "AF"))
gq_data <- read.table("GQ_per_SNP.txt", header = FALSE, col.names = c("Quality", "GQ"))

# Step 3: Check the structure of the data
glimpse(af_data)
glimpse(gq_data)
summary(af_data)
summary(gq_data)
```



```{r}
# Step 4: Create the histograms
af_plot <- ggplot(af_data, aes(x = AF)) +
  geom_histogram(binwidth = 0.05, fill = "blue", color = "black", alpha = 0.7) +  # Increased binwidth
  labs(title = "Allele Frequency Distribution per SV", x = "Allele Frequency (AF)", y = "Frequency") +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5))


gq_plot <- ggplot(gq_data, aes(x = GQ)) +
  geom_histogram(binwidth = 1, fill = "green", color = "black", alpha = 0.7) +
  labs(title = "Genotype Quality Distribution per SNP", x = "Genotype Quality (GQ)", y = "Frequency") +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5))

# Step 5: Arrange the plots side by side
grid.arrange(af_plot, gq_plot, ncol = 2)
```

```{r}
sapply(af_data, FUN = class)
summary(af_data$AF)
length(af_data$AF)#130
length(unique(af_data$AF))#130

af_data$AF_factor <- as.factor(af_data$AF)
ggplot(data = af_data, aes(x = AF_factor, y = Frequency)) +
  geom_bar(stat = "identity", width = 0.5, color="#339966", fill="#66CC99") +
  geom_text(aes(label=Frequency), vjust=-0.5, hjust = 0.5, size=2.2) +
  theme(legend.position = "none")  +
  ggtitle("Allele Frequency Distribution per SV") +
  xlab("Fréquence Allélique") + 
  ylab("Fréquence")+
  xlim("0", "1.1") +
  scale_x_discrete(breaks = levels(af_data$AF_factor)[seq(1, length(levels(af_data$AF_factor)), by = 5)])+
  theme_classic() +
  theme(axis.text.x = element_text(size = 6, angle = 45, hjust = 1)) 
 


sapply(gq_data, FUN = class)#integer et numeric
sapply(gq_data, FUN = summary)
sapply(gq_data, FUN = length)
sapply(gq_data, FUN = function(x) length(unique(x)))
length(unique(af_data$AF))
unique(gq_data$Quality)
#Quality : proba que le site n'a pas de variant : qualité de 20 : 1% de chance qu'il n'y pas de variant
#on veut que ce soit le plus haut possible


#gq_data$ <- as.factor(af_data$AF)
gq_data$GQ
gq_data$Quality_factor <- as.factor(gq_data$Quality)
ggplot(data = gq_data, aes(x = Quality_factor, y = GQ)) +
  geom_boxplot() +
  theme_classic()

names(gq_data)[names(gq_data) == "Quality"] <- "Frequency"

ggplot(data = gq_data, aes(x = GQ, y = Frequency)) +
  geom_bar(stat = "identity", width = 0.5, color="#339966", fill="#66CC99") +
  geom_text(aes(label=Frequency), vjust=-0.5, hjust = 0.5, size=2.2) +
  theme(legend.position = "none")  +
  ggtitle("Genotype Quality Distribution per SNP") +
  xlab("Genotype Quality (GQ)") + 
  ylab("Frequency")+
  #xlim("0", "1.1") +
  #scale_x_discrete(breaks = levels(af_data$AF_factor)[seq(1, length(levels(af_data$AF_factor)), by = 5)])+
  theme_classic() 
  #theme(axis.text.x = element_text(size = 6, angle = 45, hjust = 1)) 
 



```

```{r}
#Créer des groupes (breaks) pour la qualité du génotype (par exemple, de 1 à 5, 6 à 10, etc.)

gq_data$GenotypeQuality_group <- cut(gq_data$GQ,
                                  breaks = seq(1, 47, by = 1),   # Définir les breaks comme souhaité
                                  labels = c( "1-2",  "2-3",  "3-4",  "4-5",  "5-6",  "6-7",  "7-8",  "8-9",  "9-10", "10-11", "11-12", "12-13", "13-14", "14-15", "15-16", "16-17", "17-18", "18-19", "19-20", "20-21", "21-22", "22-23", "23-24", "24-25", "25-26", "26-27", "27-28", "28-29", "29-30", "30-31", "31-32", "32-33", "33-34", "34-35", "35-36", "36-37", "37-38", "38-39", "39-40", "40-41", "41-42", "42-43", "43-44", "44-45", "45-46", "46-47"), right = FALSE)


#max(gq_data$GQ) # 46.194
#min(gq_data$GQ) # 1.444

ggplot(gq_data, aes(x = GQ)) +
  geom_histogram(binwidth = 1, fill = "green", color = "black", alpha = 0.7) +
  labs(title = "Genotype Quality Distribution per SNP", x = "Genotype Quality (GQ)", y = "Frequency") +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5))

ggplot(gq_data, aes(x = GenotypeQuality_group, y = Frequency, fill = GenotypeQuality_group)) +
  geom_col() +  # Utiliser geom_col() car Frequency est déjà une valeur
  labs(title = "Fréquence en fonction de la Qualité du Génotype",
       x = "Qualité du Génotype (GQ, groupée)", y = "Fréquence") +
  theme_minimal() +
  theme(legend.position = "none", axis.text.x = element_text(angle = 45, hjust = 1))

ggplot(gq_data, aes(x = GQ)) +
  geom_histogram(binwidth = 2, fill = "#66CC99", color = "black") +  
  labs(title = "Fréquence en fonction de la Qualité du Génotype",
       x = "Qualité du Génotype (GQ)", y = "Fréquence") +
  theme_minimal()
```

# Analyse des variants (premier tableau envoyé par Teo)

```{r}
variants_P15 <- read.csv("count_P15.csv" , header = TRUE, sep = ",",  na ="NA", stringsAsFactors = FALSE)
variants_P30 <- read.csv("count_P30.csv" , header = FALSE, sep = ",",  na ="NA", stringsAsFactors = FALSE)
variants_P50 <- read.csv("count_P50.csv" , header = FALSE, sep = ",",  na ="NA", stringsAsFactors = FALSE)
variants_P65 <- read.csv("count_P65.csv" , header = FALSE, sep = ",",  na ="NA", stringsAsFactors = FALSE)
variants_P90 <- read.csv("count_90.csv" , header = FALSE, sep = ",",  na ="NA", stringsAsFactors = FALSE)


str(variants_P15)
summary(variants_P15)
sapply(variants_P15, FUN = class)

F_colnames <- function(tableau_variant){
  names(tableau_variant)[1] <- "Generation"
  names(tableau_variant)[2] <- "echantillon"
  names(tableau_variant)[3] <- "nb_variant"
  names(tableau_variant)[4] <- "sv_snp"
  tableau_variant$Generation <- as.factor(tableau_variant$Generation)
  tableau_variant$echantillon <- as.factor(tableau_variant$echantillon)
  tableau_variant$sv_snp <- as.factor(tableau_variant$sv_snp)
  return(tableau_variant)
}


variants_P15 <- F_colnames(tableau_variant = variants_P15)
variants_P15 <- variants_P15[-which(variants_P15$Generation == "*_filtered"), ]
variants_P30 <- F_colnames(tableau_variant = variants_P30)
variants_P50 <- F_colnames(tableau_variant = variants_P50)
variants_P65 <- F_colnames(tableau_variant = variants_P65)
variants_P90 <- F_colnames(tableau_variant = variants_P90)

variants_P <- rbind(variants_P15, variants_P30)
variants_P <- rbind(variants_P, variants_P50)
variants_P <- rbind(variants_P, variants_P65)
variants_P <- rbind(variants_P, variants_P90)

str(variants_P)
summary(variants_P)
sapply(variants_P, FUN = class)
sapply(variants_P, FUN = levels)
```

```{r}
#Variant P c'est un Rbind de tous les tableaux 
ggplot(data = variants_P, aes(x = sv_snp, y = nb_variant, color=sv_snp)) +
  geom_boxplot(outlier.colour="black", outlier.shape=16,
             outlier.size=2, outlier.alpha = 0.5 ,notch=FALSE) +
  #geom_dotplot(binaxis='y', stackdir='center', dotsize=1) +
  geom_jitter(shape=16, position=position_jitter(0.2), alpha = 0.5) +
  scale_color_brewer(palette="Dark2") +
  facet_wrap(~Generation, scales = "free_y") + #, ncol = 5
  labs(x = "mutation SNP et SV", y = "nombre de variants") +
  ggtitle("distribution des boîtes à moustaches des mutation\n de type SNP et SV par Génération") +
  theme_classic() +
  theme(plot.title = element_text(hjust = 0.5))

ggplot(data = variants_P, aes(x = Generation, y = nb_variant, color=sv_snp)) +
  geom_boxplot(position=position_dodge(1), outlier.colour="black", outlier.shape=16,
             outlier.size=2, outlier.alpha = 0.5 ,notch=FALSE) +
  #geom_dotplot(binaxis='y', stackdir='center', dotsize=1) +
  geom_dotplot(aes(fill = sv_snp), binaxis='y', stackdir='center', position=position_dodge(), alpha = 0.5, dotsize = 0.4, show.legend = FALSE) +
  #geom_jitter(shape=16, position=position_jitter(0.2), alpha = 0.5) +
  scale_color_brewer(palette="Dark2") +
  scale_fill_brewer(palette="Dark2") + 
  labs(x = "Génération", y = "Effectif des mutations (SV et SNP)", color = "SV / SNP") +
  ggtitle("Distribution des boîtes à moustaches des mutations\n de type SV et SNP par Génération") +
  theme_classic() +
  theme(plot.title = element_text(hjust = 0.5))
```
```{r}
remove(variants_P15, variants_P30, variants_P50, variants_P65, variants_P90, af_data, af_plot, gq_plot)
remove(F_colnames)
```

#Données brutes mutations SV et SNP

```{r}
table_snp <- read.csv("snp.csv" , header = TRUE, sep = ",",  na ="NA", stringsAsFactors = FALSE)
table_sv <- read.csv("sv.csv" , header = TRUE, sep = ",",  na ="NA", stringsAsFactors = FALSE)

names(table_snp)
names(table_snp)[names(table_snp) == "ECHANTILLONQUALITE"] <- "ECHANTILLON"
names(table_snp)[names(table_snp) == "X"] <- "QUALITE"

#table SV
str(table_sv)
sapply(table_sv, FUN = class)
table_sv$ORF <- as.factor(table_sv$ORF)
table_sv$GENERATION <- as.factor(table_sv$GENERATION)
table_sv$TYPE.DE.FICHIER <- as.factor(table_sv$TYPE.DE.FICHIER)
table_sv$TYPE.DE.MUTATION <- as.factor(table_sv$TYPE.DE.MUTATION)

summary(table_sv)
sapply(table_sv, FUN = levels)

table_sv[which(table_sv$ORF == "."),]


table_sv[which(table_sv$GENERATION == "P15"),]$ECHANTILLON %>% unique #10  3  4  5  6  7  8  9
table_sv[which(table_sv$GENERATION == "P30"),]$ECHANTILLON %>% unique #10  1  2  3  4  5  6  7  8  9
table_sv[which(table_sv$GENERATION == "P50"),]$ECHANTILLON %>% unique #10  1  2  3  4  5  6  7  8  9
table_sv[which(table_sv$GENERATION == "P65"),]$ECHANTILLON %>% unique # 1 2 3 7
table_sv[which(table_sv$GENERATION == "P90"),]$ECHANTILLON %>% unique #10  1  2  3  4  5  6  7  8  9

summary(table_sv$FREQUENCE.ALL)
```

## Les SV
compter le nb d'insertion et le nombre de deletion par échantillon et génération
on peut faire le % d'insertion aussi
faire leur longueur moyenne, medianne avec les max, les min les sq
rajouter traitement chaud, traitement froid et pas de traitement

```{r}
#rajouter traitement chaud, traitement froid et pas de traitement
table_sv <- table_sv %>% 
  mutate(traitement_type = case_when(GENERATION == "P15"~"non_traite", 
                                     (GENERATION != "P15" & ECHANTILLON <=5) ~ "traitement_froid",
                                     (GENERATION != "P15" & ECHANTILLON > 5) ~ "traitement_chaud"))
table_sv %>% names()

test <- table_sv %>% 
  group_by(GENERATION, ECHANTILLON, TYPE.DE.MUTATION) %>%
  count()
  #pivot_wider(names_from = TYPE.DE.MUTATION)


names(test)[names(test) == "n"] <- "effectif_type_mutation"

test <- left_join(test, unique(table_sv[,c("GENERATION", "ECHANTILLON", "traitement_type")]), by = c("ECHANTILLON", "GENERATION")) 



test2 <- test %>% aggregate(effectif_type_mutation ~ GENERATION + ECHANTILLON, FUN = sum)
names(test2)[names(test2) == "effectif_type_mutation"] <- "effectif_mutation"
test <- left_join(test, test2, by = c("GENERATION", "ECHANTILLON"))
remove(test2)
test <- test %>% mutate(freq_mutation_type = effectif_type_mutation/effectif_mutation)
test$freq_mutation_type <- round(test$freq_mutation_type, 3)

#frequence allélique par type de mutation
table_sv %>% names()
df <- table_sv %>% aggregate(FREQUENCE.ALL ~ GENERATION + ECHANTILLON + TYPE.DE.MUTATION, FUN = mean)
df1 <- table_sv %>% aggregate(FREQUENCE.ALL ~ GENERATION + ECHANTILLON + TYPE.DE.MUTATION, FUN = median)
df2 <- table_sv %>% aggregate(FREQUENCE.ALL ~ GENERATION + ECHANTILLON + TYPE.DE.MUTATION, FUN = sd)
df3 <- table_sv %>% aggregate(FREQUENCE.ALL ~ GENERATION + ECHANTILLON + TYPE.DE.MUTATION, FUN = max)
df4 <- table_sv %>% aggregate(FREQUENCE.ALL ~ GENERATION + ECHANTILLON + TYPE.DE.MUTATION, FUN = min)

names(df)[names(df) == "FREQUENCE.ALL"] <- "FREQUENCE.ALL_mean"
names(df1)[names(df1) == "FREQUENCE.ALL"] <- "FREQUENCE.ALL_median"
names(df2)[names(df2) == "FREQUENCE.ALL"] <- "FREQUENCE.ALL_sd"
names(df3)[names(df3) == "FREQUENCE.ALL"] <- "FREQUENCE.ALL_max"
names(df4)[names(df4) == "FREQUENCE.ALL"] <- "FREQUENCE.ALL_min"

df <- left_join(df, df1, by = c("GENERATION", "ECHANTILLON", "TYPE.DE.MUTATION"))
df <- left_join(df, df2, by = c("GENERATION", "ECHANTILLON", "TYPE.DE.MUTATION"))
df <- left_join(df, df3, by = c("GENERATION", "ECHANTILLON", "TYPE.DE.MUTATION"))
df <- left_join(df, df4, by = c("GENERATION", "ECHANTILLON", "TYPE.DE.MUTATION"))
test <- left_join(test, df, by = c("GENERATION", "ECHANTILLON", "TYPE.DE.MUTATION"))

remove(df, df1, df2, df3, df4)
#longueur du type de mutation par generation et échantillon
table_sv %>% names()
df <- table_sv %>% aggregate(LONGUEUR ~ GENERATION + ECHANTILLON + TYPE.DE.MUTATION, FUN = mean)
df1 <- table_sv %>% aggregate(LONGUEUR ~ GENERATION + ECHANTILLON + TYPE.DE.MUTATION, FUN = median)
df2 <- table_sv %>% aggregate(LONGUEUR ~ GENERATION + ECHANTILLON + TYPE.DE.MUTATION, FUN = sd)
df3 <- table_sv %>% aggregate(LONGUEUR ~ GENERATION + ECHANTILLON + TYPE.DE.MUTATION, FUN = max)
df4 <- table_sv %>% aggregate(LONGUEUR ~ GENERATION + ECHANTILLON + TYPE.DE.MUTATION, FUN = min)

names(df)[names(df) == "LONGUEUR"] <- "LONGUEUR_mean"
names(df1)[names(df1) == "LONGUEUR"] <- "LONGUEUR_median"
names(df2)[names(df2) == "LONGUEUR"] <- "LONGUEUR_sd"
names(df3)[names(df3) == "LONGUEUR"] <- "LONGUEUR_max"
names(df4)[names(df4) == "LONGUEUR"] <- "LONGUEUR_min"

df <- left_join(df, df1, by = c("GENERATION", "ECHANTILLON", "TYPE.DE.MUTATION"))
df <- left_join(df, df2, by = c("GENERATION", "ECHANTILLON", "TYPE.DE.MUTATION"))
df <- left_join(df, df3, by = c("GENERATION", "ECHANTILLON", "TYPE.DE.MUTATION"))
df <- left_join(df, df4, by = c("GENERATION", "ECHANTILLON", "TYPE.DE.MUTATION"))
test <- left_join(test, df, by = c("GENERATION", "ECHANTILLON", "TYPE.DE.MUTATION"))
remove(df, df1, df2, df3, df4)

#nb de positions différentes concernées par generation par echantillon
table_sv$POSITION_factor <- as.factor(table_sv$POSITION)
nb_positions_gen_ech <- table_sv %>% aggregate(POSITION_factor ~ GENERATION + ECHANTILLON, FUN = function(x) length(unique(x)))
names(nb_positions_gen_ech)[names(nb_positions_gen_ech) == "POSITION_factor"] <- "nb_pos_gen_ech"
test <- left_join(test, nb_positions_gen_ech, by = c("GENERATION", "ECHANTILLON")) 
remove(nb_positions_gen_ech)

#nb de positions différentes concernées par generation par echantillon par type de mutation
nb_positions_gen_ech_type_mut <- table_sv %>% aggregate(POSITION_factor ~ GENERATION + ECHANTILLON + TYPE.DE.MUTATION, FUN = function(x) length(unique(x)))
names(nb_positions_gen_ech_type_mut)[names(nb_positions_gen_ech_type_mut) == "POSITION_factor"] <- "nb_pos_gen_ech_type_mut"
test <- left_join(test, nb_positions_gen_ech_type_mut, by = c("GENERATION", "ECHANTILLON", "TYPE.DE.MUTATION")) 
remove(nb_positions_gen_ech_type_mut)

test %>% names()
test <- test %>% relocate(traitement_type, .after = ECHANTILLON)
```

```{r}
test %>% names()
ggplot(data = test, aes(x = TYPE.DE.MUTATION, y = effectif_type_mutation, color=TYPE.DE.MUTATION)) +
  geom_boxplot(outlier.colour="black", outlier.shape=16,
             outlier.size=2, outlier.alpha = 0.5 ,notch=FALSE) +
  #geom_dotplot(binaxis='y', stackdir='center', dotsize=1) +
  geom_jitter(shape=16, position=position_jitterdodge(0.2), alpha = 0.5) +
  scale_color_brewer(palette="Dark2") +
  labs(x = "Type de mutation des VS\n (délétion ou insertion)", y = "Effectif des mutations", color = "DEL = délétion /\n INS = insértion") +
  facet_wrap(~GENERATION, scales = "free_y") + #, ncol = 5
  ggtitle("Distribution des boîtes à moustaches des délétions et\n des insértions de variants structuraux (VS) par génération") +
  theme_classic()+
  theme(plot.title = element_text(hjust = 0.5))

#Même graphe qu'au dessus mais avec geom_dotplot au lien de geom_jitter
ggplot(data = test, aes(x = TYPE.DE.MUTATION, y = effectif_type_mutation, color=TYPE.DE.MUTATION)) +
  geom_boxplot(outlier.colour="black", outlier.shape=16,
             outlier.size=2, outlier.alpha = 0.5 ,notch=FALSE) +
  geom_dotplot(aes(fill = TYPE.DE.MUTATION), binaxis='y', stackdir='center',position=position_dodge(), dotsize=1, alpha = 0.5, show.legend = FALSE) +
  scale_color_brewer(palette="Dark2") +
  scale_fill_brewer(palette="Dark2") +
  labs(x = "Type de mutation des VS\n (délétion ou insertion)", y = "Effectif des mutations", color = "DEL = délétion /\n INS = insértion") +
  facet_wrap(~GENERATION, scales = "free_y") + #, ncol = 5
  ggtitle("Distribution des boîtes à moustaches des délétions et\n des insértions de variants structuraux (VS) par génération") +
  theme_classic()+
  theme(plot.title = element_text(hjust = 0.5))

ggplot(data = test, aes(x = GENERATION, y = effectif_type_mutation, color=TYPE.DE.MUTATION)) +
  geom_boxplot(position=position_dodge(1), outlier.colour="black", outlier.shape=16,
             outlier.size=2, outlier.alpha = 0.5 ,notch=FALSE) +
  geom_dotplot(aes(fill = TYPE.DE.MUTATION), binaxis='y', stackdir='center', position=position_dodge(), alpha = 0.5, dotsize = 0.5, show.legend = FALSE) +
  #geom_jitter(shape=16, position=position_jitter(), alpha = 0.5) +
  scale_color_brewer(palette="Dark2") +
  scale_fill_brewer(palette="Dark2") + 
  labs(x = "Génération", y = "Effectif des types de mutations (délétions ou insertions)", color = "DEL = délétion /\n INS = insértion") +
  ggtitle("Distribution des boîtes à moustaches des délétions et\n des insértions de variants structuraux  par génération") +
  theme_bw()+
  theme(plot.title = element_text(hjust = 0.5))
```


(j'essaye de trouver la meilleure manière de représenter visuellement la distribution statistique des insertions et des délétions)

```{r}
ggplot(data = test, aes(x = GENERATION, y = effectif_type_mutation, color=traitement_type, by = traitement_type)) +
  geom_boxplot(position=position_dodge(1),  outlier.colour="black", outlier.shape=16, outlier.size=2, outlier.alpha = 0.9 , notch=FALSE,width = 0.5) +
  geom_dotplot(aes(fill = TYPE.DE.MUTATION), binaxis='y', stackdir='center', position=position_dodge(1), alpha = 0.9, dotsize = 0.6, show.legend = TRUE) +
  #geom_jitter(aes(fill = traitement_type, shape = TYPE.DE.MUTATION) , position=position_jitterdodge(), alpha = 0.9) +
  #geom_point(aes(fill = TYPE.DE.MUTATION, shape = TYPE.DE.MUTATION), position=position_jitterdodge(dodge.width=0.5), alpha = 0.5) +
  scale_color_brewer(palette="Dark2") +
  scale_fill_manual(values = c("#990099", "#CC9900")) + 
  labs(x = "Génération", y = "Effectif des types de mutations (délétions ou insertions)", color = "Type de traitement") +
  ggtitle("Distribution des boîtes à moustaches des délétions et\n des insértions de variants structuraux  par type de traitement et par génération") +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5))

ggplot(data = test, aes(x = GENERATION, y = effectif_type_mutation, color=traitement_type)) +
  geom_boxplot(position=position_dodge(1), outlier.colour="black", outlier.shape=16,
             outlier.size=2, outlier.alpha = 0.5 ,notch=FALSE) +
  #geom_dotplot(aes(fill = TYPE.DE.MUTATION), binaxis='y', stackdir='center', position=position_dodge(), alpha = 0.5, dotsize = 0.5, show.legend = TRUE) +
  geom_jitter(aes(fill = traitement_type, shape = TYPE.DE.MUTATION), position=position_jitterdodge(), alpha = 0.9) +
  scale_color_brewer(palette="Dark2") +
  scale_fill_brewer(palette="Dark2") + 
  labs(x = "Génération", y = "Effectif des types de mutations\n(délétions ou insertions)", color = "Type de traitement") +
  ggtitle("Distribution des boîtes à moustaches des délétions et des\ninsértions de variants structuraux  par type de traitement et par génération") +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5))


#C'est la même chose qu'au dessus mais dans des boites pour les générations
ggplot(data = test, aes(x = traitement_type, y = effectif_type_mutation, color=traitement_type, by = traitement_type)) +
  geom_boxplot(position=position_dodge(1), outlier.colour="black", outlier.shape=16, outlier.size=2, outlier.alpha = 0.5 , notch=FALSE, width = 0.5) +
  geom_dotplot(aes(fill = TYPE.DE.MUTATION), binaxis='y', stackdir='center', position=position_dodge(), alpha = 0.8, dotsize = 1) +
  #geom_jitter(aes(fill = traitement_type, shape = TYPE.DE.MUTATION) , position=position_jitterdodge(), alpha = 0.8) +
  scale_color_brewer(palette="Dark2") +
  scale_fill_manual(values = c("#990099", "#CC9900")) + 
  facet_wrap(~GENERATION, scales = "free_y") +
  labs(x = "Type de traitement", y = "Effectif des types de mutations\n(délétions ou insertions)", color = "Type de traitement") +
  ggtitle("Distribution des boîtes à moustaches des délétions et des\ninsértions de variants structuraux  par type de traitement et par génération") +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5)) +
  theme(axis.text.x = element_text(size = 6, angle = 45, hjust = 1)) 

ggplot(data = test, aes(x = traitement_type, y = effectif_type_mutation, color=traitement_type, by = traitement_type)) +
  geom_boxplot(position=position_dodge(1), outlier.colour="black", outlier.shape=16, outlier.size=2, outlier.alpha = 0.5 , notch=FALSE, width = 0.5) +
  #geom_dotplot(aes(fill = TYPE.DE.MUTATION, shape = TYPE.DE.MUTATION), binaxis='y', stackdir='center', position=position_dodge(), alpha = 0.5, dotsize = 0.3) +
  geom_jitter(aes(fill = traitement_type, shape = TYPE.DE.MUTATION) , position=position_jitterdodge(), alpha = 0.8) +
  scale_color_brewer(palette="Dark2") +
  facet_wrap(~GENERATION, scales = "free_y") +
  labs(x = "Type de traitement", y = "Effectif des types de mutations\n(délétions ou insertions)", color = "Type de traitement") +
  ggtitle("Distribution des boîtes à moustaches des délétions et des\ninsértions de variants structuraux  par type de traitement et par génération") +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5)) +
  theme(axis.text.x = element_text(size = 6, angle = 45, hjust = 1)) 
```

```{r}

```


```{r}
test[,c("GENERATION", "traitement_type", "ECHANTILLON", "effectif_mutation")] %>% unique() %>%
  ggplot( aes(x = GENERATION, 
                        y = effectif_mutation, 
                        color=traitement_type, 
                        by = traitement_type)) +
  geom_boxplot(position=position_dodge(1), 
               outlier.colour="black", 
               outlier.shape=16,
               outlier.size=2, 
               outlier.alpha = 0.5 ,
               notch=FALSE,
               width = 0.5) +
  geom_jitter(aes(fill = traitement_type) , position=position_jitterdodge(), alpha = 0.9) +
  scale_color_brewer(palette="Dark2") +
  labs(x = "Génération", y = "mutations") +
  theme_bw()

test[,c("GENERATION", "traitement_type", "ECHANTILLON", "effectif_mutation")] %>% unique() %>%
  ggplot(aes(x = GENERATION,
           y = effectif_mutation,
           by = traitement_type,
           fill = traitement_type)) +
  geom_bar(position=position_dodge(), width=0.5, stat = "identity")+
  scale_fill_brewer(palette="Dark2") +
  theme_bw()

table_sv$TYPE.DE.MUTATION
table_sv[,c("GENERATION", "traitement_type", "TYPE.DE.MUTATION")] %>%
  ggplot(aes(x = traitement_type,
           by = TYPE.DE.MUTATION,
           fill = TYPE.DE.MUTATION)) +
  geom_bar(position=position_stack(), width=0.5, stat = "count")+
  scale_fill_brewer(palette="Dark2") +
  facet_wrap(~GENERATION, scale = "free") +
  theme_bw()


ggplot(test, aes(x = traitement_type, y = freq_mutation_type, by = TYPE.DE.MUTATION, fill = TYPE.DE.MUTATION)) +
  geom_bar(stat = "identity", position = "fill", width= 0.8, color = "black") +
  facet_wrap(~GENERATION, scale = "free") +
  #geom_text(aes(label = pourcent_eff), stat = "identity",  position = position_fill(.5), size = 2.5, color = "black") +  #facultatif sinon on ne voit rien
  scale_fill_brewer(palette="Dark2") +
  geom_text(aes(label = freq_mutation_type), stat = "identity",  position = position_fill(.5), size = 2, color = "black") +  
    theme_bw()

#test[,c("GENERATION", "traitement_type", "ECHANTILLON", "effectif_mutation")] %>% unique() %>%
  ggplot(test, aes(x = traitement_type, 
                        y = freq_mutation_type, 
                        color=TYPE.DE.MUTATION, 
                        by = TYPE.DE.MUTATION)) +
  geom_boxplot(position=position_dodge(1), 
               outlier.colour="black", 
               outlier.shape=16,
               outlier.size=2, 
               outlier.alpha = 0.5 ,
               notch=FALSE,
               width = 0.5) +
  geom_jitter(aes(fill = traitement_type) , position=position_jitterdodge(), alpha = 0.9) +
  facet_wrap(~GENERATION, scale = "free") +  
  scale_color_brewer(palette="Dark2") +
  labs(x = "Génération", y = "mutations") +
  theme_bw()
```


```{r}

test[which(test$GENERATION == "P90"), ] %>%
  ggplot( aes(x = traitement_type, 
                        y = effectif_type_mutation, 
                        color=TYPE.DE.MUTATION, 
                        by = TYPE.DE.MUTATION)) +
  
  geom_boxplot(position=position_dodge(1), 
               outlier.colour="black", 
               outlier.shape=16,
               outlier.size=2, 
               outlier.alpha = 0.5 ,
               notch=FALSE,
               width = 0.5) +
  #geom_dotplot(aes(fill = TYPE.DE.MUTATION, shape = TYPE.DE.MUTATION), binaxis='y', stackdir='center', position=position_dodge(), alpha = 0.5, dotsize = 0.3) +
  geom_jitter(aes(fill = TYPE.DE.MUTATION) , position=position_jitterdodge(), alpha = 0.7) +
  #geom_point(aes(fill = TYPE.DE.MUTATION, shape = TYPE.DE.MUTATION), position=position_jitterdodge(dodge.width=0.5), alpha = 0.5) +
  scale_color_brewer(palette="Dark2") +
  labs(x = "type de traitement", y = "effectif des type de mutations") +
  ggtitle("P90") +
  theme_bw()

test[which(test$GENERATION == "P65"), ] %>%
  ggplot( aes(x = traitement_type, 
                        y = effectif_type_mutation, 
                        color=TYPE.DE.MUTATION, 
                        by = TYPE.DE.MUTATION)) +
  
  geom_boxplot(position=position_dodge(1), 
               outlier.colour="black", 
               outlier.shape=16,
               outlier.size=2, 
               outlier.alpha = 0.5 ,
               notch=FALSE,
               width = 0.5) +
  #geom_dotplot(aes(fill = TYPE.DE.MUTATION, shape = TYPE.DE.MUTATION), binaxis='y', stackdir='center', position=position_dodge(), alpha = 0.5, dotsize = 0.3) +
  geom_jitter(aes(fill = TYPE.DE.MUTATION) , position=position_jitterdodge(), alpha = 0.7) +
  #geom_point(aes(fill = TYPE.DE.MUTATION, shape = TYPE.DE.MUTATION), position=position_jitterdodge(dodge.width=0.5), alpha = 0.5) +
  scale_color_brewer(palette="Dark2") +
  labs(x = "type de traitement", y = "effectif des type de mutations") +
  ggtitle("P65") +
  theme_bw()

test[which(test$GENERATION == "P50"), ] %>%
  ggplot( aes(x = traitement_type, 
                        y = effectif_type_mutation, 
                        color=TYPE.DE.MUTATION, 
                        by = TYPE.DE.MUTATION)) +
  
  geom_boxplot(position=position_dodge(1), 
               outlier.colour="black", 
               outlier.shape=16,
               outlier.size=2, 
               outlier.alpha = 0.5 ,
               notch=FALSE,
               width = 0.5) +
  #geom_dotplot(aes(fill = TYPE.DE.MUTATION, shape = TYPE.DE.MUTATION), binaxis='y', stackdir='center', position=position_dodge(), alpha = 0.5, dotsize = 0.3) +
  geom_jitter(aes(fill = TYPE.DE.MUTATION) , position=position_jitterdodge(), alpha = 0.7) +
  #geom_point(aes(fill = TYPE.DE.MUTATION, shape = TYPE.DE.MUTATION), position=position_jitterdodge(dodge.width=0.5), alpha = 0.5) +
  scale_color_brewer(palette="Dark2") +
  labs(x = "type de traitement", y = "effectif des type de mutations") +
  ggtitle("P50") +
  theme_bw()

test[which(test$GENERATION == "P30"), ] %>%
  ggplot( aes(x = traitement_type, 
                        y = effectif_type_mutation, 
                        color=TYPE.DE.MUTATION, 
                        by = TYPE.DE.MUTATION)) +
  
  geom_boxplot(position=position_dodge(1), 
               outlier.colour="black", 
               outlier.shape=16,
               outlier.size=2, 
               outlier.alpha = 0.5 ,
               notch=FALSE,
               width = 0.5) +
  #geom_dotplot(aes(fill = TYPE.DE.MUTATION, shape = TYPE.DE.MUTATION), binaxis='y', stackdir='center', position=position_dodge(), alpha = 0.5, dotsize = 0.3) +
  geom_jitter(aes(fill = TYPE.DE.MUTATION) , position=position_jitterdodge(), alpha = 0.7) +
  #geom_point(aes(fill = TYPE.DE.MUTATION, shape = TYPE.DE.MUTATION), position=position_jitterdodge(dodge.width=0.5), alpha = 0.5) +
  scale_color_brewer(palette="Dark2") +
  labs(x = "type de traitement", y = "effectif des type de mutations") +
  ggtitle("P30") +
  theme_bw()

test[which(test$GENERATION == "P15"), ] %>%
  ggplot( aes(x = traitement_type, 
                        y = effectif_type_mutation, 
                        color=TYPE.DE.MUTATION, 
                        by = TYPE.DE.MUTATION)) +
  
  geom_boxplot(position=position_dodge(1), 
               outlier.colour="black", 
               outlier.shape=16,
               outlier.size=2, 
               outlier.alpha = 0.5 ,
               notch=FALSE,
               width = 0.5) +
  #geom_dotplot(aes(fill = TYPE.DE.MUTATION, shape = TYPE.DE.MUTATION), binaxis='y', stackdir='center', position=position_dodge(), alpha = 0.5, dotsize = 0.3) +
  geom_jitter(aes(fill = TYPE.DE.MUTATION) , position=position_jitterdodge(), alpha = 0.7) +
  #geom_point(aes(fill = TYPE.DE.MUTATION, shape = TYPE.DE.MUTATION), position=position_jitterdodge(dodge.width=0.5), alpha = 0.5) +
  scale_color_brewer(palette="Dark2") +
  labs(x = "type de traitement", y = "effectif des type de mutations") +
  ggtitle("P15") +
  theme_bw()
```

Regarder la longueur des mutations en fonction de si c'est del ou ins et en fonction du traitement et des génération

```{r}
table_genome_ann_CyHV3 <- read.csv("annotation_genes_CyHV3_genbank.csv", header = TRUE, sep = "\t")
table_genome_ann_CyHV3 %>% names()
table_genome_ann_CyHV3 <- mutate(table_genome_ann_CyHV3, gen_length = End-Begin)
table_genome_ann_CyHV3 <- table_genome_ann_CyHV3 %>% relocate (gen_length, .after = End)
table_genome_ann_CyHV3$gen_length %>% summary()
table_genome_ann_CyHV3$Gene.Type %>% unique() #protein-coding

#   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
#    267    1018    1969    2694    3205   30495 
```


regarder les positions du génome les plus touchées par les mutations 
```{r}
table_sv <- table_sv[order(table_sv$POSITION),] 
#regarder le nombre de mutations par positions 

# Génome : 295100 pb
genome_ref <- seq(1, 295100, 100)

gq_data$GenotypeQuality_group <- cut(gq_data$GQ,
                                  breaks = seq(1, 47, by = 1),   # Définir les breaks comme souhaité
                                  labels = c( "1-2",  "2-3",  "3-4",  "4-5",  "5-6",  "6-7",  "7-8",  "8-9",  "9-10", "10-11", "11-12", "12-13", "13-14", "14-15", "15-16", "16-17", "17-18", "18-19", "19-20", "20-21", "21-22", "22-23", "23-24", "24-25", "25-26", "26-27", "27-28", "28-29", "29-30", "30-31", "31-32", "32-33", "33-34", "34-35", "35-36", "36-37", "37-38", "38-39", "39-40", "40-41", "41-42", "42-43", "43-44", "44-45", "45-46", "46-47"), right = FALSE)
```


