---
title: "GQ_AF_markdown.R"
author: "heloise_calzan"
date: "2025-02-13"
output: pdf_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```


```{r cars}
setwd("E:\\github\\BILL_2025\\R")


# Step 1: Load necessary libraries
library(tidyverse)
library(gridExtra)
library(dplyr)

library(rstatix)
```



Palettes de couleurs R : 
https://bookdown.org/hneth/ds4psy/D.3-apx-colors-basics.html
https://derekogle.com/NCGraphing/resources/colors
```{r color palette}
c25 <- c( "dodgerblue2", "#E31A1C", "green4", "#6A3D9A", "#FF7F00", "black", "gold1", "skyblue2", "#FB9A99",  "palegreen2", "#CAB2D6", "#FDBF6F", "gray70", "khaki2", "maroon", "orchid1", "deeppink1", "blue1", "steelblue4","darkturquoise", "green1", "yellow4", "yellow3","darkorange4", "brown", "#E69F00", "#56B4E9", "#009E73", "#F0E442", "#0072B2", "#D55E00", "#CC79A7", "#000000", "#E69F00", "#56B4E9", "#009E73", "#F0E442", 'black','forestgreen', 'red2', 'orange', 'cornflowerblue',  'magenta', 'darkolivegreen4', 'indianred1', 'tan4', 'darkblue', 'mediumorchid1','firebrick4',  'yellowgreen', 'lightsalmon', 'tan3', "tan1",'darkgray', 'wheat4', '#DDAD4B', 'chartreuse','seagreen1', 'moccasin', 'mediumvioletred', 'seagreen','cadetblue1',"darkolivegreen1" ,"tan2" ,   "tomato3" , "#7CE3D8","gainsboro")

library(Polychrome)
library(pals)
# install.packages("rcartocolor")
# install.packages("remotes")
library(rcartocolor)
pal.bands(alphabet, alphabet2, cols25, glasbey, kelly, polychrome, 
  stepped, tol, watlington,
  show.names=FALSE)
library(wesanderson)
#carto_pal

scales::show_col(carto_pal(12, "Safe"))
              

```

# Analyse préliminaires Genotype quality & Allelic frequency

```{r cars}
# Step 2: Import the data
af_data <- read.table(".\\fichiers_utilises\\AF_per_SV.txt", header = FALSE, col.names = c("Frequency", "AF"))
gq_data <- read.table(".\\fichiers_utilises\\GQ_per_SNP.txt", header = FALSE, col.names = c("Quality", "GQ"))

# Step 3: Check the structure of the data
glimpse(af_data)
glimpse(gq_data)
summary(af_data)
summary(gq_data)
```



```{r}
# Step 4: Create the histograms
af_plot <- ggplot(af_data, aes(x = AF)) +
  geom_histogram(binwidth = 0.05, fill = "blue", color = "black", alpha = 0.7) +  # Increased binwidth
  labs(title = "Allele Frequency Distribution per SV", x = "Allele Frequency (AF)", y = "Frequency") +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5))


gq_plot <- ggplot(gq_data, aes(x = GQ)) +
  geom_histogram(binwidth = 1, fill = "green", color = "black", alpha = 0.7) +
  labs(title = "Genotype Quality Distribution per SNP", x = "Genotype Quality (GQ)", y = "Frequency") +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5))

# Step 5: Arrange the plots side by side
grid.arrange(af_plot, gq_plot, ncol = 2)
```

```{r}
sapply(af_data, FUN = class)
summary(af_data$AF)
length(af_data$AF)#130
length(unique(af_data$AF))#130

af_data$AF_factor <- as.factor(af_data$AF)
ggplot(data = af_data, aes(x = AF_factor, y = Frequency)) +
  geom_bar(stat = "identity", width = 0.5, color="#339966", fill="#66CC99") +
  geom_text(aes(label=Frequency), vjust=-0.5, hjust = 0.5, size=2.2) +
  theme(legend.position = "none")  +
  ggtitle("Allele Frequency Distribution per SV") +
  xlab("Fréquence Allélique") + 
  ylab("Fréquence")+
  xlim("0", "1.1") +
  scale_x_discrete(breaks = levels(af_data$AF_factor)[seq(1, length(levels(af_data$AF_factor)), by = 5)])+
  theme_classic() +
  theme(axis.text.x = element_text(size = 6, angle = 45, hjust = 1)) 
 


sapply(gq_data, FUN = class)#integer et numeric
sapply(gq_data, FUN = summary)
sapply(gq_data, FUN = length)
sapply(gq_data, FUN = function(x) length(unique(x)))
length(unique(af_data$AF))
unique(gq_data$Quality)
#Quality : proba que le site n'a pas de variant : qualité de 20 : 1% de chance qu'il n'y pas de variant
#on veut que ce soit le plus haut possible


#gq_data$ <- as.factor(af_data$AF)
gq_data$GQ
gq_data$Quality_factor <- as.factor(gq_data$Quality)
ggplot(data = gq_data, aes(x = Quality_factor, y = GQ)) +
  geom_boxplot() +
  theme_classic()

names(gq_data)[names(gq_data) == "Quality"] <- "Frequency"

ggplot(data = gq_data, aes(x = GQ, y = Frequency)) +
  geom_bar(stat = "identity", width = 0.5, color="#339966", fill="#66CC99") +
  geom_text(aes(label=Frequency), vjust=-0.5, hjust = 0.5, size=2.2) +
  theme(legend.position = "none")  +
  ggtitle("Genotype Quality Distribution per SNP") +
  xlab("Genotype Quality (GQ)") + 
  ylab("Frequency")+
  #xlim("0", "1.1") +
  #scale_x_discrete(breaks = levels(af_data$AF_factor)[seq(1, length(levels(af_data$AF_factor)), by = 5)])+
  theme_classic() 
  #theme(axis.text.x = element_text(size = 6, angle = 45, hjust = 1)) 
 



```

```{r}
#Créer des groupes (breaks) pour la qualité du génotype (par exemple, de 1 à 5, 6 à 10, etc.)

gq_data$GenotypeQuality_group <- cut(gq_data$GQ,
                                  breaks = seq(1, 47, by = 1),   # Définir les breaks comme souhaité
                                  labels = c( "1-2",  "2-3",  "3-4",  "4-5",  "5-6",  "6-7",  "7-8",  "8-9",  "9-10", "10-11", "11-12", "12-13", "13-14", "14-15", "15-16", "16-17", "17-18", "18-19", "19-20", "20-21", "21-22", "22-23", "23-24", "24-25", "25-26", "26-27", "27-28", "28-29", "29-30", "30-31", "31-32", "32-33", "33-34", "34-35", "35-36", "36-37", "37-38", "38-39", "39-40", "40-41", "41-42", "42-43", "43-44", "44-45", "45-46", "46-47"), right = FALSE)


#max(gq_data$GQ) # 46.194
#min(gq_data$GQ) # 1.444
gq_data$GQ %>% max()

ggplot(gq_data, aes(x = GQ)) +
  geom_histogram(binwidth = 1, fill = "green", color = "black", alpha = 0.7) +
  labs(title = "Genotype Quality Distribution per SNP", x = "Genotype Quality (GQ)", y = "Frequency") +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5))

ggplot(gq_data, aes(x = GenotypeQuality_group, y = Frequency, fill = GenotypeQuality_group)) +
  geom_col() +  # Utiliser geom_col() car Frequency est déjà une valeur
  labs(title = "Fréquence en fonction de la Qualité du Génotype",
       x = "Qualité du Génotype (GQ, groupée)", y = "Fréquence") +
  theme_minimal() +
  theme(legend.position = "none", axis.text.x = element_text(angle = 45, hjust = 1))

gq_plot_2.0 <- ggplot(gq_data, aes(x = GQ)) +
  geom_histogram(binwidth = 2, fill = "aquamarine3", color = "aquamarine4", alpha = 0.8) +  
  labs(title = "Fréquence en fonction de la Qualité du Génotype",
       x = "Qualité du Génotype (GQ)", y = "Fréquence") +
  scale_x_continuous(breaks = seq(1, 47, by = 2)) +
  theme_minimal()
gq_plot_2.0
```

```{r}
#ggsave(plot = gq_plot_2.0, ".\\images_scripts\\gqplot.png")

remove(af_data, af_plot, gq_data, gq_plot, gq_plot_2.0)
```


#Données brutes mutations SV et SNP

```{r}
table_nb_variants <- read.csv(".\\fichiers_utilises\\counts.csv" , header = TRUE, sep = ";",  na ="NA", stringsAsFactors = FALSE)

table_snp <- read.csv(".\\fichiers_utilises\\snp.csv" , header = TRUE, sep = ",",  na ="NA", stringsAsFactors = FALSE)
table_sv <- read.csv(".\\fichiers_utilises\\sv.csv" , header = TRUE, sep = ",",  na ="NA", stringsAsFactors = FALSE)

names(table_snp)
names(table_snp)[names(table_snp) == "ECHANTILLONQUALITE"] <- "ECHANTILLON"
names(table_snp)[names(table_snp) == "X"] <- "QUALITE"
```

```{r, include=FALSE}
#count_snp
table_snp %>% names()
count_snp <- table_snp %>% group_by(GEN, ECH) %>% count()
count_snp %>% names()
names(count_snp)[names(count_snp) == "n"] <- "COUNT"
names(count_snp)[names(count_snp) == "GEN"] <- "GENERATION"
names(count_snp)[names(count_snp) == "ECH"] <- "ECHANTILLON"

table_nb_variants$SV_SNP <- as.factor("SV")
count_snp$SV_SNP <- as.factor("SNP")
count_snp$TYPE <- as.factor("SUB")#pour substitution

#count_SV
table_sv %>% names()
count_ins_del_sv <- table_sv %>% group_by(GENERATION, ECHANTILLON, TYPE.DE.MUTATION) %>% count()
names(count_ins_del_sv)[names(count_ins_del_sv) == "n"] <- "COUNT_TYPE"
names(count_ins_del_sv)[names(count_ins_del_sv) == "TYPE.DE.MUTATION"] <- "TYPE"




count_ins_del_sv$SV_SNP <- as.factor("SV")
table_nb_variants %>% names()
count_ins_del_sv %>% names()
table_nb_variants %>% nrow()
count_ins_del_sv %>% nrow()
count_ins_del_sv <- left_join(count_ins_del_sv, table_nb_variants, by = c("GENERATION" , "ECHANTILLON" , "SV_SNP"))

remove(table_nb_variants)


count_snp <- count_snp %>% mutate(COUNT_TYPE = COUNT)
count_snp %>% names()
count_ins_del_sv %>% names()
count_ins_del_sv <- count_ins_del_sv %>% relocate(COUNT, .after = ECHANTILLON)
count_snp <- count_snp %>% relocate(SV_SNP, .after = COUNT_TYPE)

count_snp_sv_type <- rbind(count_snp, count_ins_del_sv)


#ajout d'une colonne traitement : 
count_snp_sv_type <- count_snp_sv_type %>% 
  mutate(TRAITEMENT = case_when(GENERATION == "P15"~"aucun", 
                                     (GENERATION != "P15" & ECHANTILLON <=5) ~ "Froid",
                                     (GENERATION != "P15" & ECHANTILLON > 5) ~ "Chaud"))
count_snp_sv_type <- count_snp_sv_type %>% relocate(TRAITEMENT, .after = ECHANTILLON)



nb_mutation_ech_gen <- count_snp_sv_type %>% unique() %>% aggregate (COUNT ~ GENERATION + ECHANTILLON, FUN = function(x) sum(unique(x)))
names(nb_mutation_ech_gen)[names(nb_mutation_ech_gen) == "COUNT"] <- "COUNT_ech_gen"
count_snp_sv_type <- left_join(count_snp_sv_type, nb_mutation_ech_gen)
remove(nb_mutation_ech_gen)


#je rajoute cette colonne dans la table SV
table_sv <- table_sv %>% 
  mutate(TRAITEMENT = case_when(GENERATION == "P15"~"aucun", 
                                     (GENERATION != "P15" & ECHANTILLON <=5) ~ "Froid",
                                     (GENERATION != "P15" & ECHANTILLON > 5) ~ "Chaud"))

```

Je vais travailler sur : count_snp_sv_type
```{r, include=FALSE}
# rajouter la colonne COUNT_GEN
count_snp_sv_type[, c("GENERATION", "COUNT_ech_gen", "ECHANTILLON")] %>% unique() %>%
  ggplot(aes(x = GENERATION, y = COUNT_ech_gen, color = as.factor(ECHANTILLON))) +
  geom_point() +
  geom_line() +
  #scale_fill_brewer(palette="Dark2") +
  #scale_color_brewer(palette="Dark2") +
  labs(x = "Génération", y = "Effectif total") +
  ggtitle("nombre de mutations total dans les échantillons par génération") +
  theme_classic() +
  theme(axis.title.y = element_text(angle=0, vjust = 0.5))


count_snp_sv_type[, c("GENERATION", "COUNT_ech_gen", "ECHANTILLON")] %>% unique() %>%
  ggplot(aes(x = GENERATION, y = COUNT_ech_gen, fill = as.factor(ECHANTILLON))) +
  geom_bar(position = position_dodge(1), stat = "identity") +
  #scale_fill_brewer(palette="Dark2") +
  #scale_color_brewer(palette="Dark2") +
  labs(x = "Génération", y = "Effectif total") +
  ggtitle("nombre de mutations total dans les échantillons par génération") +
  theme_classic() +
  theme(axis.title.y = element_text(angle=0, vjust = 0.5))
```


```{r}
nb_mu_moy <- count_snp_sv_type %>%  aggregate(COUNT_ech_gen ~ GENERATION, FUN = mean)
nb_mu_moy <- nb_mu_moy %>% mutate(GEN_num = case_when(nb_mu_moy$GENERATION == "P15"~ 15, 
                                                      nb_mu_moy$GENERATION == "P30"~ 30, 
                                                      nb_mu_moy$GENERATION == "P50"~ 50, 
                                                      nb_mu_moy$GENERATION == "P65"~ 65, 
                                                      nb_mu_moy$GENERATION == "P90"~ 90, 
                                                      ))
nb_mu_moy$GEN_num <- as.numeric(nb_mu_moy$GEN_num)
P_nb_mu_1 <- nb_mu_moy %>%
  ggplot(aes(x = GEN_num, y = COUNT_ech_gen)) +
  geom_point() +
  geom_line() +
  labs(x = "Génération", y = "Effectif\ntotal\nmoyen") +
  ggtitle("Nombre moyen de mutations par génération") +
  theme_classic() +
  theme(axis.title.y = element_text(angle=0, vjust = 0.5))


nb_mu_moy_t <- count_snp_sv_type %>%  aggregate(COUNT_ech_gen ~ GENERATION + TRAITEMENT, FUN = mean)
nb_mu_moy_t <- nb_mu_moy_t %>% mutate(GEN_num = case_when(nb_mu_moy_t$GENERATION == "P15"~ 15, 
                                                      nb_mu_moy_t$GENERATION == "P30"~ 30, 
                                                      nb_mu_moy_t$GENERATION == "P50"~ 50, 
                                                      nb_mu_moy_t$GENERATION == "P65"~ 65, 
                                                      nb_mu_moy_t$GENERATION == "P90"~ 90, 
                                                      ))
nb_mu_moy_t$GEN_num <- as.numeric(nb_mu_moy_t$GEN_num)
P_nb_mu_2 <- nb_mu_moy_t %>%
  ggplot(aes(x = GEN_num, y = COUNT_ech_gen, color = TRAITEMENT)) +
  geom_point() +
  geom_line() +
  labs(x = "Génération", y = "Effectif\ntotal\nmoyen") +
  ggtitle("Nombre moyen de mutations par génération et par traitement") +
  scale_color_brewer(palette="Dark2") +
  theme_classic() +
  theme(axis.title.y = element_text(angle=0, vjust = 0.5))

#ggsave(plot = P_nb_mu_1, ".\\images_scripts\\nb_moy_mutation_Gen.png", height = 5, width = 7)
#ggsave(plot = P_nb_mu_2, ".\\images_scripts\\nb_moy_mutation_Gen_T.png", height = 5, width = 7)
P_nb_mu_1
P_nb_mu_2
#remove(nb_mu_moy, nb_mu_moy_t)
```


```{r}
count_snp_sv_type %>%
  ggplot(aes(x = GENERATION, y = COUNT_TYPE, color= TYPE)) +
  geom_point() +
  #ggtitle("Distribution du nombre de mutation par Génération\n(un point = un échantillon") +
  theme_bw() +
  labs(title = "Distribution du nombre de mutation par Génération",
              subtitle = "un point = un échantillon")

count_snp_sv_type[which(count_snp_sv_type$SV_SNP == "SV"), ] %>%
  ggplot(aes(x = GENERATION, y = COUNT_TYPE, color= TYPE)) +
  geom_point() +
  labs(title = "Distribution du nombre d'indels par Génération",
              subtitle = "un point = un échantillon") +
  theme_bw()
```


```{r}
#statistiques du nb de mutation en fonction du traitement
count_snp_sv_type[, c("GENERATION","ECHANTILLON" ,"COUNT_ech_gen", "TRAITEMENT")] %>% unique() %>%
  ggplot(aes(x = GENERATION, y = COUNT_ech_gen)) +
  geom_boxplot(outlier.colour="black", position = position_dodge(1) ,outlier.shape=16, outlier.size=2, outlier.alpha = 0.5 ,notch=FALSE) +
  geom_dotplot( position = position_dodge(1), binaxis='y', stackdir='center', dotsize=0.3, alpha = 0.7) +
   #geom_jitter(shape=16, position=position_jitter(0.2), alpha = 0.5) 
  scale_fill_brewer(palette="Dark2") +
  scale_color_brewer(palette="Dark2") +
  labs(x = "Génération", y = "Effectif") +
  ggtitle("Distribution des statistiques de nomnbre de mutations en fonction de la génération") +
  theme_classic() +
  theme(axis.title.y = element_text(angle=0, vjust = 0.5))

count_snp_sv_type[, c("GENERATION", "ECHANTILLON", "COUNT_ech_gen", "TRAITEMENT")] %>% unique() %>%
  ggplot(aes(x = GENERATION, y = COUNT_ech_gen, color= TRAITEMENT)) +
  geom_boxplot(outlier.colour="black", position = position_dodge(1) ,outlier.shape=16, outlier.size=2, outlier.alpha = 0.5 ,notch=FALSE) +
  geom_dotplot(aes(fill = TRAITEMENT) , position = position_dodge(1), binaxis='y', stackdir='center', dotsize=0.3, alpha = 0.7) +
   #geom_jitter(shape=16, position=position_jitter(0.2), alpha = 0.5) 
  scale_fill_brewer(palette="Dark2") +
  scale_color_brewer(palette="Dark2") +
  labs(x = "Génération", y = "Effectif") +
  ggtitle("Distribution des statistiques de nomnbre de mutations en fonction du traitement et de la génération") +
  theme_classic() +
  theme(axis.title.y = element_text(angle=0, vjust = 0.5))

count_snp_sv_type[, c("GENERATION", "ECHANTILLON", "COUNT_ech_gen", "TRAITEMENT")] %>% unique() %>%
  ggplot(aes(x = GENERATION, y = COUNT_ech_gen, color= TRAITEMENT)) +
  geom_boxplot(outlier.colour="black", position = position_dodge(1) ,outlier.shape=16, outlier.size=2, outlier.alpha = 0.5 ,notch=FALSE) +
  geom_dotplot(aes(fill = TRAITEMENT) , position = position_dodge(1), binaxis='y', stackdir='center', dotsize=0.3, alpha = 0.7) +
   #geom_jitter(shape=16, position=position_jitter(0.2), alpha = 0.5) 
  scale_fill_brewer(palette="Dark2") +
  scale_color_brewer(palette="Dark2") +
  labs(x = "Génération", y = "Effectif") +
  ggtitle("Distribution des statistiques de nomnbre de mutations en fonction du traitement et de la génération") +
  theme_classic() +
  theme(axis.title.y = element_text(angle=0, vjust = 0.5))


```


```{r}
count_snp_sv_type[,c("GENERATION", "ECHANTILLON", "COUNT", "SV_SNP")] %>% unique() %>%
 # filter(TYPE %in% c("SUB", "INS")) %>%
  ggplot(aes(x = GENERATION, y = COUNT, color= SV_SNP)) +
  geom_boxplot(outlier.colour="black", outlier.shape=16, outlier.size=2, outlier.alpha = 0.5 ,notch=FALSE, position = position_dodge(1)) +
  geom_dotplot(aes(fill = SV_SNP) ,binaxis='y', stackdir='center', dotsize=0.5, alpha = 0.7, position = position_dodge(1)) +
   #geom_jitter(shape=16, position=position_jitter(0.2), alpha = 0.5) 
  scale_fill_brewer(palette="Dark2") +
  scale_color_brewer(palette="Dark2") +
  #facet_wrap(~ GENERATION)  +  #, ncol = 5
  labs(x = "mutation SNP et SV", y = "nombre de variants") +
  ggtitle("Statistiques des mutation de type SNP et SV par Génération") +
  theme_classic()

count_snp_sv_type[,c("GENERATION", "ECHANTILLON", "COUNT", "SV_SNP", "TRAITEMENT")] %>% unique() %>%
  ggplot(aes(x = GENERATION, y = COUNT, color= TRAITEMENT)) +
  geom_boxplot(outlier.colour="black", outlier.shape=16, outlier.size=2, outlier.alpha = 0.5 ,notch=FALSE, position = position_dodge(1)) +
  geom_dotplot(aes(fill = TRAITEMENT) ,binaxis='y', stackdir='center', dotsize=0.5, alpha = 0.7, position = position_dodge(1)) +
   #geom_jitter(shape=16, position=position_jitter(0.2), alpha = 0.5) 
  scale_fill_brewer(palette="Dark2",  name = "Traitement", labels = c("Aucun", "Chaud", "Froid")) +
  scale_color_brewer(palette="Dark2", name = "Traitement", labels = c("Aucun", "Chaud", "Froid")) +
  facet_wrap(~ SV_SNP, scales = "free")  +  #, ncol = 5
  labs(x = "Générations", y = "Effectif") +
  ggtitle("Statistiques des mutation de type SNP et SV par Génération et traitement") +
  theme_classic() +
  theme(axis.title.y = element_text(angle = 0, vjust = 0.5))
#ggsave(".\\images_scripts\\boxplot_nb_snp_sv_TG.png", height = 5, width = 7)

count_snp_sv_type[,c("GENERATION", "ECHANTILLON", "COUNT", "SV_SNP", "TRAITEMENT")] %>% unique() %>%
  ggplot(aes(x = GENERATION,
           y = COUNT,
           by = TRAITEMENT,
           fill = TRAITEMENT)) +
  geom_bar(position=position_dodge(), width=0.5, stat = "identity")+
  scale_fill_brewer(palette="Dark2", name = "Traitement", labels = c("Aucun", "Chaud", "Froid")) +
  #scale_fill_discrete(name = "Traitement", labels = c("Aucun", "Chaud", "Froid"))+
  labs(x = "Génération", y = "Effectif", fill = "Type de traitement") +
  facet_wrap(~ SV_SNP, scales = "free")  + 
  ggtitle("Nombre de mutations (SNP ou Structural Variant) par traitement\net par génération") +
  theme_bw()+
  theme(axis.title.y = element_text(angle = 0, vjust = 0.5))

#ggsave(".\\images_scripts\\nb_snp_sv_TG.png", height = 5, width = 7)
```



```{r}
count_snp_sv_type %>% 
  ggplot(aes(x = GENERATION, y = COUNT_TYPE, color= TYPE)) +
  geom_boxplot(outlier.colour="black", outlier.shape=16, outlier.size=2, outlier.alpha = 0.5 ,notch=FALSE, position = position_dodge(1)) +
  geom_dotplot(aes(fill = TYPE) ,binaxis='y', stackdir='center', dotsize=0.5, alpha = 0.7, position = position_dodge(1)) +
   #geom_jitter(shape=16, position=position_jitter(0.2), alpha = 0.5) 
  scale_fill_brewer(palette="Dark2") +
  scale_color_brewer(palette="Dark2") +
  #facet_wrap(~ GENERATION)  +  #, ncol = 5
  labs(x = "Génération", y = "nombre de mutation") +
  ggtitle("Statistiques des types de mutation par Génération") +
  theme_classic() 

count_snp_sv_type %>% 
  ggplot(aes(x = GENERATION, y = COUNT_TYPE, color= TYPE)) +
  geom_boxplot(outlier.colour="black", outlier.shape=16, outlier.size=2, outlier.alpha = 0.5 ,notch=FALSE, position = position_dodge(1)) +
  geom_dotplot(aes(fill = TYPE) ,binaxis='y', stackdir='center', dotsize=0.5, alpha = 0.7, position = position_dodge(1)) +
   #geom_jitter(shape=16, position=position_jitter(0.2), alpha = 0.5) 
  scale_fill_brewer(palette="Dark2") +
  scale_color_brewer(palette="Dark2") +
  facet_wrap(~ TYPE, scales = "free")  +  #, ncol = 5
  labs(x = "Génération", y = "nombre de mutation", title = "distribution des statistiques des types mutations par Génération") +
  theme_classic()

#ggsave(".\\images_scripts\\P_nb_variant_generation.png", width = 14, height = 6)
```


```{r, width = 8, height = 14}
plot_boxplot_snpsvEff <-count_snp_sv_type %>% 
  ggplot(aes(x = GENERATION, y = COUNT_TYPE, color= TRAITEMENT)) +
  geom_boxplot(outlier.colour="black", position = position_dodge(1) ,outlier.shape=16, outlier.size=2, outlier.alpha = 0.5 ,notch=FALSE) +
  geom_dotplot(aes(fill = TRAITEMENT) , position = position_dodge(1), binaxis='y', stackdir='center', dotsize=0.5, alpha = 0.7) +
   #geom_jitter(shape=16, position=position_jitter(0.2), alpha = 0.5) 
  scale_fill_brewer(palette="Dark2") +
  scale_color_brewer(palette="Dark2") +
  facet_wrap(~ TYPE, scales = "free", ncol = 1)  +  #, ncol = 5
  labs(x = "Génération", y = "Effectif") +
  ggtitle("Statistiques des types de mutations en fonction du traitement et de la génération") +
  theme_classic() +
  theme(axis.title.y = element_text(angle=0, vjust = 0.5))

plot_boxplot_snpsvEff2 <-count_snp_sv_type %>% 
  ggplot(aes(x = GENERATION, y = COUNT_TYPE, color= TRAITEMENT)) +
  geom_boxplot(outlier.colour="black", position = position_dodge(1) ,outlier.shape=16, outlier.size=2, outlier.alpha = 0.5 ,notch=FALSE) +
  geom_dotplot(aes(fill = TRAITEMENT) , position = position_dodge(1), binaxis='y', stackdir='center', dotsize=0.3, alpha = 0.7) +
   #geom_jitter(shape=16, position=position_jitter(0.2), alpha = 0.5) 
  scale_fill_brewer(palette="Dark2") +
  scale_color_brewer(palette="Dark2") +
  facet_wrap(~ TYPE, scales = "free")  +  #, ncol = 5
  labs(x = "Génération", y = "Effectif") +
  ggtitle("Distribution des statistiques des types de mutations en fonction du traitement et de la génération") +
  theme_classic() +
  theme(axis.title.y = element_text(angle=0, vjust = 0.5))

plot_boxplot_snpsvEff
plot_boxplot_snpsvEff2

#ggsave(plot = plot_boxplot_snpsvEff, ".\\images_scripts\\plot_boxplot_snpsvEff.png", width = 8, height = 12)
#ggsave(plot = plot_boxplot_snpsvEff2, ".\\images_scripts\\plot_boxplot_snpsvEff2.png", width = 14, height = 6)
```



```{r tests stat, include=FALSE}
#Comparer plusieurs générations entre elles : 
#kruskal_test(count_snp_sv_type[which(count_snp_sv_type$TYPE == "SUB"),], COUNT_TYPE ~ GENERATION)
count_snp_sv_type %>%
  filter(TYPE == "SUB") %>%
  kruskal.test(COUNT_TYPE ~ GENERATION) # Kruskal-Wallis chi-squared = 415.16, df = 8, p-value < 2.2e-16

count_snp_sv_type %>%
  filter(TYPE == "INS") %>%
  kruskal.test(COUNT_TYPE ~ GENERATION)
#Kruskal-Wallis chi-squared = 313.07, df = 8, p-value < 2.2e-16

count_snp_sv_type %>%
  filter(TYPE == "DEL") %>%
  kruskal.test(COUNT_TYPE ~ GENERATION)
#Kruskal-Wallis chi-squared = 220.91, df = 8, p-value < 2.2e-16


#Si la p-value est < 0.05, il y a une différence significative entre les générations.
#Tu peux ensuite faire un test post-hoc pour voir quelles générations diffèrent :
count_snp_sv_type$GENERATION <- as.factor(count_snp_sv_type$GENERATION)

count_snp_sv_type %>%
  filter(TYPE == "SUB", 
         GENERATION == "P90") %>% nrow() #10, #10, #10, #10, 10
count_snp_sv_type$COUNT_TYPE <- as.numeric(count_snp_sv_type$COUNT_TYPE)

# Filtrer uniquement les générations P15 et P30 -> SUBSTITUTIONS
df_test <- count_snp_sv_type %>% filter(TYPE == "SUB", GENERATION %in% c("P15", "P30")) #%>%
pairwise.wilcox.test(df_test$COUNT_TYPE, df_test$GENERATION, p.adjust.method = "bonferroni") # 0.12

df_test <- count_snp_sv_type %>% filter(TYPE == "SUB", GENERATION %in% c("P15", "P50")) 
pairwise.wilcox.test(df_test$COUNT_TYPE, df_test$GENERATION, p.adjust.method = "bonferroni") # 0.14

df_test <- count_snp_sv_type %>% filter(TYPE == "SUB", GENERATION %in% c("P15", "P65"))
pairwise.wilcox.test(df_test$COUNT_TYPE, df_test$GENERATION, p.adjust.method = "bonferroni") # 0.82

df_test <- count_snp_sv_type %>% filter(TYPE == "SUB", GENERATION %in% c("P15", "P90"))
pairwise.wilcox.test(df_test$COUNT_TYPE, df_test$GENERATION, p.adjust.method = "bonferroni") # 0.54

df_test <- count_snp_sv_type %>% filter(TYPE == "SUB", GENERATION %in% c("P30", "P50"))
pairwise.wilcox.test(df_test$COUNT_TYPE, df_test$GENERATION, p.adjust.method = "bonferroni") # 0.82

df_test <- count_snp_sv_type %>% filter(TYPE == "SUB", GENERATION %in% c("P30", "P65"))
pairwise.wilcox.test(df_test$COUNT_TYPE, df_test$GENERATION, p.adjust.method = "bonferroni") #0.00088

df_test <- count_snp_sv_type %>% filter(TYPE == "SUB", GENERATION %in% c("P30", "P90"))
pairwise.wilcox.test(df_test$COUNT_TYPE, df_test$GENERATION, p.adjust.method = "bonferroni") #0.00018

df_test <- count_snp_sv_type %>% filter(TYPE == "SUB", GENERATION %in% c("P50", "P65"))
pairwise.wilcox.test(df_test$COUNT_TYPE, df_test$GENERATION, p.adjust.method = "bonferroni") #0.00033

df_test <- count_snp_sv_type %>% filter(TYPE == "SUB", GENERATION %in% c("P50", "P90"))
pairwise.wilcox.test(df_test$COUNT_TYPE, df_test$GENERATION, p.adjust.method = "bonferroni") #0.00018

df_test <- count_snp_sv_type %>% filter(TYPE == "SUB", GENERATION %in% c("P65", "P90"))
pairwise.wilcox.test(df_test$COUNT_TYPE, df_test$GENERATION, p.adjust.method = "bonferroni") #0.57

# INSERTIONS
df_test <- count_snp_sv_type %>% filter(TYPE == "INS", GENERATION %in% c("P15", "P30")) 
pairwise.wilcox.test(df_test$COUNT_TYPE, df_test$GENERATION, p.adjust.method = "bonferroni") # 0.15

df_test <- count_snp_sv_type %>% filter(TYPE == "INS", GENERATION %in% c("P15", "P50")) 
pairwise.wilcox.test(df_test$COUNT_TYPE, df_test$GENERATION, p.adjust.method = "bonferroni") #0.32

df_test <- count_snp_sv_type %>% filter(TYPE == "INS", GENERATION %in% c("P15", "P65"))
pairwise.wilcox.test(df_test$COUNT_TYPE, df_test$GENERATION, p.adjust.method = "bonferroni") # 0.2

df_test <- count_snp_sv_type %>% filter(TYPE == "INS", GENERATION %in% c("P15", "P90"))
pairwise.wilcox.test(df_test$COUNT_TYPE, df_test$GENERATION, p.adjust.method = "bonferroni") # 5e-04

df_test <- count_snp_sv_type %>% filter(TYPE == "INS", GENERATION %in% c("P30", "P50"))
pairwise.wilcox.test(df_test$COUNT_TYPE, df_test$GENERATION, p.adjust.method = "bonferroni") # 0.057

df_test <- count_snp_sv_type %>% filter(TYPE == "INS", GENERATION %in% c("P30", "P65"))
pairwise.wilcox.test(df_test$COUNT_TYPE, df_test$GENERATION, p.adjust.method = "bonferroni") #0.074

df_test <- count_snp_sv_type %>% filter(TYPE == "INS", GENERATION %in% c("P30", "P90"))
pairwise.wilcox.test(df_test$COUNT_TYPE, df_test$GENERATION, p.adjust.method = "bonferroni") #0.00028

df_test <- count_snp_sv_type %>% filter(TYPE == "INS", GENERATION %in% c("P50", "P65"))
pairwise.wilcox.test(df_test$COUNT_TYPE, df_test$GENERATION, p.adjust.method = "bonferroni") #0.18

df_test <- count_snp_sv_type %>% filter(TYPE == "INS", GENERATION %in% c("P50", "P90"))
pairwise.wilcox.test(df_test$COUNT_TYPE, df_test$GENERATION, p.adjust.method = "bonferroni") #2e-04

df_test <- count_snp_sv_type %>% filter(TYPE == "INS", GENERATION %in% c("P65", "P90"))
pairwise.wilcox.test(df_test$COUNT_TYPE, df_test$GENERATION, p.adjust.method = "bonferroni") #0.78

# DELETION
df_test <- count_snp_sv_type %>% filter(TYPE == "DEL", GENERATION %in% c("P15", "P30")) 
pairwise.wilcox.test(df_test$COUNT_TYPE, df_test$GENERATION, p.adjust.method = "bonferroni") # 0.29

df_test <- count_snp_sv_type %>% filter(TYPE == "DEL", GENERATION %in% c("P15", "P50")) 
pairwise.wilcox.test(df_test$COUNT_TYPE, df_test$GENERATION, p.adjust.method = "bonferroni") #0.59

df_test <- count_snp_sv_type %>% filter(TYPE == "DEL", GENERATION %in% c("P15", "P65"))
pairwise.wilcox.test(df_test$COUNT_TYPE, df_test$GENERATION, p.adjust.method = "bonferroni") # 0.03

df_test <- count_snp_sv_type %>% filter(TYPE == "DEL", GENERATION %in% c("P15", "P90"))
pairwise.wilcox.test(df_test$COUNT_TYPE, df_test$GENERATION, p.adjust.method = "bonferroni") # 0.029

df_test <- count_snp_sv_type %>% filter(TYPE == "DEL", GENERATION %in% c("P30", "P50"))
pairwise.wilcox.test(df_test$COUNT_TYPE, df_test$GENERATION, p.adjust.method = "bonferroni") # 0.057

df_test <- count_snp_sv_type %>% filter(TYPE == "DEL", GENERATION %in% c("P30", "P65"))
pairwise.wilcox.test(df_test$COUNT_TYPE, df_test$GENERATION, p.adjust.method = "bonferroni") #0.048

df_test <- count_snp_sv_type %>% filter(TYPE == "DEL", GENERATION %in% c("P30", "P90"))
pairwise.wilcox.test(df_test$COUNT_TYPE, df_test$GENERATION, p.adjust.method = "bonferroni") #0.93

df_test <- count_snp_sv_type %>% filter(TYPE == "DEL", GENERATION %in% c("P50", "P65"))
pairwise.wilcox.test(df_test$COUNT_TYPE, df_test$GENERATION, p.adjust.method = "bonferroni") #0.02

df_test <- count_snp_sv_type %>% filter(TYPE == "DEL", GENERATION %in% c("P50", "P90"))
pairwise.wilcox.test(df_test$COUNT_TYPE, df_test$GENERATION, p.adjust.method = "bonferroni") #0.018

df_test <- count_snp_sv_type %>% filter(TYPE == "DEL", GENERATION %in% c("P65", "P90"))
pairwise.wilcox.test(df_test$COUNT_TYPE, df_test$GENERATION, p.adjust.method = "bonferroni") #0.71



#Cela compare chaque paire de générations en ajustant les p-values.

#Comparaison des traitements au sein d’une génération

#Si tu veux comparer les traitements au sein d’une même génération, utilise Wilcoxon (si 2 traitements) ou Kruskal-Wallis (si >2 traitements).
#Wilcoxon pour 2 traitements

#Si tu compares deux traitements (ex: "aucun" vs "traité") dans P15 :
df_test <- count_snp_sv_type %>% filter(TYPE == "SUB", GENERATION == "P30") 
pairwise.wilcox.test(df_test$COUNT_TYPE, df_test$TRAITEMENT, p.adjust.method = "bonferroni") #0.095
df_test <- count_snp_sv_type %>% filter(TYPE == "SUB", GENERATION == "P50") 
pairwise.wilcox.test(df_test$COUNT_TYPE, df_test$TRAITEMENT, p.adjust.method = "bonferroni") #0.21
df_test <- count_snp_sv_type %>% filter(TYPE == "SUB", GENERATION == "P65") 
pairwise.wilcox.test(df_test$COUNT_TYPE, df_test$TRAITEMENT, p.adjust.method = "bonferroni") #0.22
df_test <- count_snp_sv_type %>% filter(TYPE == "SUB", GENERATION == "P90") 
pairwise.wilcox.test(df_test$COUNT_TYPE, df_test$TRAITEMENT, p.adjust.method = "bonferroni") #1 : impossible de calculer la p-value exacte avec des ex-aequos


df_test <- count_snp_sv_type %>% filter(TYPE == "INS", GENERATION == "P30") 
pairwise.wilcox.test(df_test$COUNT_TYPE, df_test$TRAITEMENT, p.adjust.method = "bonferroni") #0.89
df_test <- count_snp_sv_type %>% filter(TYPE == "INS", GENERATION == "P50") 
pairwise.wilcox.test(df_test$COUNT_TYPE, df_test$TRAITEMENT, p.adjust.method = "bonferroni") #1
df_test <- count_snp_sv_type %>% filter(TYPE == "INS", GENERATION == "P65") 
pairwise.wilcox.test(df_test$COUNT_TYPE, df_test$TRAITEMENT, p.adjust.method = "bonferroni") #0.5
df_test <- count_snp_sv_type %>% filter(TYPE == "INS", GENERATION == "P90") 
pairwise.wilcox.test(df_test$COUNT_TYPE, df_test$TRAITEMENT, p.adjust.method = "bonferroni") #0.035



df_test <- count_snp_sv_type %>% filter(TYPE == "DEL", GENERATION == "P30") 
pairwise.wilcox.test(df_test$COUNT_TYPE, df_test$TRAITEMENT, p.adjust.method = "bonferroni") #0.48
df_test <- count_snp_sv_type %>% filter(TYPE == "DEL", GENERATION == "P50") 
pairwise.wilcox.test(df_test$COUNT_TYPE, df_test$TRAITEMENT, p.adjust.method = "bonferroni") #0.71
df_test <- count_snp_sv_type %>% filter(TYPE == "DEL", GENERATION == "P65") 
pairwise.wilcox.test(df_test$COUNT_TYPE, df_test$TRAITEMENT, p.adjust.method = "bonferroni") 
df_test <- count_snp_sv_type %>% filter(TYPE == "DEL", GENERATION == "P90") 
pairwise.wilcox.test(df_test$COUNT_TYPE, df_test$TRAITEMENT, p.adjust.method = "bonferroni") #0.047


#Si la p-value est < 0.05, il y a une différence significative entre les traitements.
remove(df_test)
```



#Est-ce que la frequence des types de mutation change au cours des génération et selon les traitements
```{r}

count_snp_sv_type <- count_snp_sv_type %>% mutate(freq_type = COUNT_TYPE/COUNT_ech_gen)
remove(df_test)
count_snp_sv_type$freq_type <-  round(count_snp_sv_type$freq_type, 3)

#frequence moyenne par génération
df_freq_moy_gen <- count_snp_sv_type %>% aggregate(freq_type ~ GENERATION + TYPE, FUN = mean)
df2 <- count_snp_sv_type %>% aggregate(freq_type ~ GENERATION + TYPE, FUN = sd)
names(df2)[names(df2) == "freq_type"] <- "freq_type_sd"
df_freq_moy_gen <- left_join(df_freq_moy_gen, df2)
remove(df2)
df_freq_moy_gen$freq_type <- round(df_freq_moy_gen$freq_type, 3)
df_freq_moy_gen$freq_type_sd <- round(df_freq_moy_gen$freq_type_sd, 3)


df_freq_moy_gen <- df_freq_moy_gen %>% mutate(GEN_num = case_when(df_freq_moy_gen$GENERATION == "P15"~ 15, 
                                                      df_freq_moy_gen$GENERATION == "P30"~ 30, 
                                                      df_freq_moy_gen$GENERATION == "P50"~ 50, 
                                                      df_freq_moy_gen$GENERATION == "P65"~ 65, 
                                                      df_freq_moy_gen$GENERATION == "P90"~ 90, 
                                                      ))

ggplot(df_freq_moy_gen, aes(x = GEN_num, y = freq_type, color = TYPE)) +
  geom_point() +
  geom_line()+
  scale_color_brewer(palette = "Dark2") +
  labs(title = "Fréquences moyennes des types de mutations dans chaque génération", y = "Fréquence", x = "Génération") +
  theme_bw()

#par génération, par traitement
df_freq_moy_gen_t <- count_snp_sv_type %>% aggregate(freq_type ~ GENERATION + TRAITEMENT + TYPE, FUN = mean)
df2 <- count_snp_sv_type %>% aggregate(freq_type ~ GENERATION + TYPE, FUN = sd)
names(df2)[names(df2) == "freq_type"] <- "freq_type_sd"
df_freq_moy_gen_t <- left_join(df_freq_moy_gen_t, df2)
remove(df2)
df_freq_moy_gen_t$freq_type <- round(df_freq_moy_gen_t$freq_type, 3)
df_freq_moy_gen_t$freq_type_sd <- round(df_freq_moy_gen_t$freq_type_sd, 3)

df_freq_moy_gen_t <- df_freq_moy_gen_t %>% mutate(GEN_num = case_when(df_freq_moy_gen_t$GENERATION == "P15"~ 15, 
                                                      df_freq_moy_gen_t$GENERATION == "P30"~ 30, 
                                                      df_freq_moy_gen_t$GENERATION == "P50"~ 50, 
                                                      df_freq_moy_gen_t$GENERATION == "P65"~ 65, 
                                                      df_freq_moy_gen_t$GENERATION == "P90"~ 90, 
                                                      ))

ggplot(df_freq_moy_gen_t, aes(x = GEN_num, y = freq_type, color = TYPE)) +
  geom_point() +
  geom_line()+
  scale_color_brewer(palette = "Dark2") +
  facet_wrap(~TRAITEMENT) +
  labs(title = "Fréquences moyennes des types de mutations dans chaque génération et par traitement", y = "Fréquence", x = "Génération") +
  theme_bw() 
  
ggplot(df_freq_moy_gen_t, aes(x = GEN_num, y = freq_type, color = TRAITEMENT)) +
  geom_point() +
  geom_line()+
  scale_color_brewer(palette = "Dark2") +
  facet_wrap(~TYPE)+
  theme_bw()

#remove(df_freq_moy_gen_traitement, df_freq_moy_gen)
```




```{r fig.height= 13, fig.width=6}
count_snp_sv_type  %>%
  ggplot(aes(x = as.factor(ECHANTILLON), y = freq_type, fill = TYPE, color = TRAITEMENT)) +
  geom_bar(aes(color = TRAITEMENT), stat = "identity", position = "fill", width= 0.8, size = 1) +
  geom_text(aes(label = freq_type), stat = "identity",  position = position_fill(.5), size = 2, color = "black") + 
  scale_fill_brewer(palette = "Set2") +
  facet_wrap(~GENERATION, ncol = 5) +
  scale_color_manual(values = c("#339966", "#990000", "#0066CC"))+ 
  labs(title = "Fréquence relative des types de mutations (snp, insertion, délétion)\ndans chaque échantillon par génération") +
  theme_bw()

ggsave(".\\images_scripts\\freq_snp_ins_del_ech_gen.png", height = 5, width = 17)


```


```{r}
#là j'ai voulu faire une moyenne de fréquence des mutations dans les générations sans prendre en compte le traitement
df_freq_moy_gen %>%
  ggplot(aes(x = as.factor(GENERATION), y = freq_type, fill = TYPE)) +
  geom_bar( stat = "identity", position = position_dodge(1)) +
  geom_text(aes(label = freq_type), stat = "identity", vjust = - 1,position = position_dodge(1) ,size = 2, color = "black")+
  geom_errorbar(aes(ymin= freq_type - freq_type_sd, ymax=freq_type + freq_type_sd), width=.2, position = position_dodge(1), na.rm = TRUE) +
  scale_fill_brewer(palette="Dark2") +
  theme_bw()

df_freq_moy_gen %>%
  ggplot(aes(x = as.factor(GENERATION), y = freq_type, fill = TYPE)) +
  geom_bar( stat = "identity", position = position_fill(0.5)) +
  geom_text(aes(label = freq_type), stat = "identity", vjust = 0, position = position_fill(0.5) , size = 2, color = "black")+
  scale_fill_brewer(palette="Dark2") +
  theme_bw()

df_freq_moy_gen %>%
  ggplot(aes(x = as.factor(GENERATION), y = freq_type, color = TYPE)) +
  geom_point() +
  geom_line(aes(group = TYPE)) +
  geom_errorbar(aes(ymin= freq_type - freq_type, ymax=freq_type + freq_type), width=.2, na.rm = TRUE) +
  scale_fill_brewer(palette="Dark2") +
  theme_bw()
```


```{r}
#Par génération, oar traitement

df_freq_moy_gen_t %>%
  ggplot(aes(x = as.factor(GENERATION), y = freq_type, fill = TRAITEMENT)) +
  geom_bar( stat = "identity", position = position_dodge(1)) +
  geom_text(aes(label = freq_type), stat = "identity", vjust = - 1,position = position_dodge(1) ,size = 2, color = "black")+
  geom_errorbar(aes(ymin= freq_type - freq_type_sd, ymax=freq_type + freq_type_sd), width=.2, position = position_dodge(1), na.rm = TRUE) +
  scale_fill_brewer(palette="Dark2") +
  facet_wrap(~TYPE) +
  theme_bw()

df_freq_moy_gen_t %>%
  ggplot(aes(x = as.factor(GENERATION), y = freq_type, fill = TYPE)) +
  geom_bar( stat = "identity", position = position_dodge(1)) +
  geom_text(aes(label = freq_type), stat = "identity", vjust = 0, position = position_dodge(1) , size = 2, color = "black")+
  geom_errorbar(aes(ymin= freq_type - freq_type_sd, ymax=freq_type + freq_type_sd), width=.2, position = position_dodge(1), na.rm = TRUE) +
  scale_fill_brewer(palette="Dark2") +
  facet_wrap(~TRAITEMENT) +
  #scale_y_log10()+
  theme_bw()


```
```{r}
remove(df_freq_moy_gen, df_freq_moy_gen_t)
```



##Distribution des fréquences relatives par échantillon pour les SV
```{r}
count_ins_del_sv <- count_ins_del_sv %>% 
  mutate(TRAITEMENT = case_when(GENERATION == "P15" ~ "aucun", 
                                GENERATION != "P15" & ECHANTILLON <=5 ~ "Froid",
                                GENERATION != "P15" & ECHANTILLON > 5 ~ "Chaud")) %>%
  relocate(TRAITEMENT, .after = ECHANTILLON)

count_ins_del_sv <- count_ins_del_sv %>% mutate(freq_indel = COUNT_TYPE/COUNT)


count_ins_del_sv$freq_indel <- round(count_ins_del_sv$freq_indel, 3)
count_ins_del_sv <- count_ins_del_sv %>% mutate(GEN_TRAITEMENT = paste(as.factor(GENERATION), as.factor(TRAITEMENT), sep = "_"))
count_ins_del_sv %>% names()


  
count_ins_del_sv %>%
  ggplot(aes(x = as.factor(ECHANTILLON), y = freq_indel, by = TYPE, fill = TYPE)) +
  geom_bar(stat = "identity", position = "fill", width= 0.8, color = "black") +
  geom_text(aes(label = freq_indel), stat = "identity",  position = position_fill(.5), size = 2, color = "black") + 
  scale_fill_brewer(palette = "Set2") +
  facet_wrap(~GEN_TRAITEMENT) +
  labs(title = "Fréquence relative des délétions et des subsitution dans chaque échantillon par génération") +
  theme_bw()

count_ins_del_sv %>%
  ggplot(aes(x = as.factor(ECHANTILLON), y = freq_indel, fill = TYPE, color = TRAITEMENT)) +
  geom_bar(aes(color = TRAITEMENT), stat = "identity", position = "fill", width= 0.8, size = 1) +
  geom_text(aes(label = freq_indel), stat = "identity",  position = position_fill(.5), size = 2, color = "black") + 
  scale_fill_brewer(palette = "Set2") +
  facet_wrap(~GENERATION) +
  scale_color_manual(values = c("#339966", "#0066CC", "#990000"))+
  labs(title = "Fréquence relative des délétions et des subsitution dans chaque échantillon par génération") +
  theme_bw()

count_ins_del_sv %>%
  ggplot(aes(x = as.factor(ECHANTILLON), y = freq_indel, by = TYPE, fill = TYPE)) +
  geom_bar(stat = "identity", position = "fill", width= 0.8, color = "black") +
  geom_text(aes(label = freq_indel), stat = "identity",  position = position_fill(.5), size = 2, color = "black") + 
  scale_fill_brewer(palette = "Set2") +
  facet_wrap(~GEN_TRAITEMENT, scales = "free") +
  labs(title = "Fréquence relative des délétions et des subsitution dans chaque échantillon par génération") +
  theme_bw()
  

```


```{r fig.height=6, fig.width=15}
count_ins_del_sv  %>%
  ggplot(aes(x = as.factor(ECHANTILLON), y = freq_indel, fill = TYPE, color = TRAITEMENT)) +
  geom_bar(aes(color = TRAITEMENT), stat = "identity", position = "fill", width= 0.8, size = 1) +
  geom_text(aes(label = freq_indel), stat = "identity",  position = position_fill(.5), size = 2, color = "black") + 
  scale_fill_brewer(palette = "Set2") +
  facet_wrap(~GENERATION, ncol = 5) +
  #scale_fill_discrete(name = "Type de SV", labels = c("Insertion", "Délétion")) +
  #scale_color_discrete(name = "Traitement" , labels("Aucun", "Chaud", "Froid")) +
  scale_color_manual(values = c("#339966", "#990000", "#0066CC"), name = "Traitement", labels("Aucun", "Chaud", "Froid"))+ 
  scale_fill_brewer(palette = "Set2", name = "Type de SV") +
  labs(title = "Fréquence relative des délétions et des subsitution dans chaque échantillon par génération") +
  labs(y = "Fréquence", x = "Echantillon") +
  theme_bw() +
  theme(axis.title.y = element_text(angle = 0, vjust = 0.5), legend.position = "bottom")

ggsave(".\\images_scripts\\freq_ins_del_ech_gen.png", height = 5, width = 17)

```



```{r, fig.height = 7, fig.width = 13 }
count_ins_del_sv %>%
  ggplot(aes(x = as.factor(ECHANTILLON), y = freq_indel, fill = TYPE, color = TRAITEMENT)) +
  geom_bar(aes(color = TRAITEMENT), stat = "identity", position = "fill", width= 0.8, size = 1) +
  geom_text(aes(label = freq_indel), stat = "identity",  position = position_fill(.5), size = 2, color = "black") + 
  scale_fill_brewer(palette = "Set2") +
  facet_wrap(~GENERATION, ncol = 3) +
  scale_color_manual(values = c("#339966", "#990000", "#0066CC"))+
  labs(title = "Fréquence relative des délétions et des insertions dans chaque échantillon par génération") +
  theme_bw()

ggsave(".\\images_scripts\\freq_del_ins_ech.png", height = 6, width = 10)
```




```{r}
#là j'ai voulu faire une moyenne de fréquence des indels dans les générations sans prendre en compte le traitement

freq_moy_indel_gen <- count_ins_del_sv %>% aggregate(freq_indel ~ GENERATION + TYPE, FUN = mean)
df2 <- count_ins_del_sv %>% aggregate(freq_indel ~ GENERATION + TYPE, FUN = sd)
names(df2)[names(df2) == "freq_indel"] <- "freq_indel_sd"
freq_moy_indel_gen <- left_join(freq_moy_indel_gen, df2)
remove(df2)
freq_moy_indel_gen$freq_indel<- round(freq_moy_indel_gen$freq_indel, 3)
freq_moy_indel_gen$freq_indel_sd<- round(freq_moy_indel_gen$freq_indel_sd, 3)

freq_moy_indel_gen <- freq_moy_indel_gen %>% mutate(GEN_num = case_when(freq_moy_indel_gen$GENERATION == "P15"~ 15, 
                                                      freq_moy_indel_gen$GENERATION == "P30"~ 30, 
                                                      freq_moy_indel_gen$GENERATION == "P50"~ 50, 
                                                      freq_moy_indel_gen$GENERATION == "P65"~ 65, 
                                                      freq_moy_indel_gen$GENERATION == "P90"~ 90, 
                                                      ))


freq_moy_indel_gen %>%
  ggplot(aes(x = as.factor(GENERATION), y = freq_indel, fill = TYPE)) +
  geom_bar( stat = "identity", position = position_dodge(1)) +
  geom_text(aes(label = freq_indel), stat = "identity", vjust = - 1,position = position_dodge(1) ,size = 2, color = "black")+
  geom_errorbar(aes(ymin= freq_indel - freq_indel_sd, ymax=freq_indel + freq_indel_sd), width=.2, position = position_dodge(1), na.rm = TRUE) +
 # scale_fill_discrete(name = "Dose", labels = c("A", "B", "C"))
  scale_fill_brewer(palette="Dark2", name = "Type de\nmutation", labels = c("Délétion", "Insertion")) +
  labs(x = "Génération", y = "Fréquence\nmoyenne", title = "Fréquences moyennes des insertions et\ndes délétions par génération") +
  theme_bw() +
  theme(axis.title.y = element_text(angle = 0, vjust = 0.5))
#ggsave(".\\images_scripts\\freq_ins_del_gen.png", height = 4, width = 6)

freq_moy_indel_gen %>%
  ggplot(aes(x = as.factor(GENERATION), y = freq_indel, color = TYPE)) +
  geom_point() +
  geom_line(aes(group = TYPE)) +
  geom_errorbar(aes(ymin= freq_indel - freq_indel_sd, ymax=freq_indel + freq_indel_sd), width=.2, na.rm = TRUE) +
  scale_fill_brewer(palette="Dark2") +
  theme_bw()

```


```{r}
nb_indel_ech_gen_t <- count_ins_del_sv %>% aggregate(freq_indel ~ GENERATION + TRAITEMENT + TYPE , FUN = mean)
df2 <- count_ins_del_sv %>% aggregate(freq_indel ~ GENERATION + TRAITEMENT + TYPE, FUN = sd)
names(df2)[names(df2) == "freq_indel"] <- "freq_indel_sd"
nb_indel_ech_gen_t <- left_join(nb_indel_ech_gen_t, df2)
remove(df2)

nb_indel_ech_gen_t$freq_indel <- round(nb_indel_ech_gen_t$freq_indel, 3)
nb_indel_ech_gen_t$freq_indel_sd <- round(nb_indel_ech_gen_t$freq_indel_sd, 3)

nb_indel_ech_gen_t <- nb_indel_ech_gen_t %>% mutate(GEN_num = case_when(nb_indel_ech_gen_t$GENERATION == "P15"~ 15, 
                                                      nb_indel_ech_gen_t$GENERATION == "P30"~ 30, 
                                                      nb_indel_ech_gen_t$GENERATION == "P50"~ 50, 
                                                      nb_indel_ech_gen_t$GENERATION == "P65"~ 65, 
                                                      nb_indel_ech_gen_t$GENERATION == "P90"~ 90, 
                                                      ))



nb_indel_ech_gen_t %>%
  ggplot(aes(x = as.factor(GENERATION), y = freq_indel, fill = TRAITEMENT)) +
  geom_bar( stat = "identity", position = position_dodge(1), whidth = 0.5 ) +
  geom_text(aes(label = freq_indel), stat = "identity", vjust = 0, position = position_dodge(1) , size = 2, color = "black")+
  geom_errorbar(aes(ymin= freq_indel - freq_indel_sd, ymax=freq_indel + freq_indel_sd), width=.2, position = position_dodge(1), na.rm = TRUE) +
  scale_fill_brewer(palette="Dark2") +
  facet_wrap(~TYPE ) +
  theme_bw()

nb_indel_ech_gen_t %>%
  ggplot(aes(x = as.factor(GENERATION), y = freq_indel, fill = TYPE)) +
  geom_bar( stat = "identity", position = position_dodge(1), whidth = 0.5 ) +
  geom_text(aes(label = freq_indel), stat = "identity", vjust = 0, position = position_dodge(1) , size = 2, color = "black")+
  geom_errorbar(aes(ymin= freq_indel - freq_indel_sd, ymax=freq_indel + freq_indel_sd), width=.2, position = position_dodge(1), na.rm = TRUE) +
  scale_fill_brewer(palette="Dark2") +
  facet_wrap(~TRAITEMENT ) +
  theme_bw()





nb_indel_ech_gen_t %>%
  ggplot(aes(x = as.factor(GENERATION), y = freq_indel, color = TRAITEMENT)) +
  geom_point()+
  geom_line(aes(group = TRAITEMENT))+
  geom_text(aes(label = freq_indel), stat = "identity", vjust = - 1,position = position_dodge(1) ,size = 2, color = "black")+
  #geom_errorbar(aes(ymin= freq_indel - freq_indel_sd, ymax=freq_indel + freq_indel_sd), width=.2, na.rm = TRUE) +
  scale_color_brewer(palette="Dark2") +
  facet_wrap(~TYPE) +
  theme_bw()


```



```{r}
count_ins_del_sv[,c("GENERATION", "TRAITEMENT", "ECHANTILLON", "COUNT")] %>% unique() %>%
  ggplot(aes(x = GENERATION,
           y = COUNT,
           by = TRAITEMENT,
           fill = TRAITEMENT)) +
  geom_bar(position=position_dodge(), width=0.5, stat = "identity")+
  scale_fill_brewer(palette="Dark2") +
  labs(x = "Génération", y = "Effectif des SV", fill = "Type de traitement") +
   ggtitle("Distribution des mutations de variants structuraux\n(délétions et des insértions confondues)   par type de traitement et par génération") +
  theme_bw()

count_ins_del_sv[,c("GENERATION", "TRAITEMENT", "COUNT_TYPE", "TYPE")] %>% unique() %>%
  ggplot(aes(x = GENERATION,
           y = COUNT_TYPE,
           by = TRAITEMENT,
           fill = TRAITEMENT)) +
  geom_bar(position=position_dodge(), width=0.5, stat = "identity")+
  scale_fill_brewer(palette="Dark2") +
  facet_wrap(~TYPE)+
  labs(x = "Génération", y = "Effectif des SV", fill = "Type de traitement") +
   ggtitle("Distribution des mutations de variants structuraux\npar type de traitement et par génération") +
  theme_bw()
```


on a un tableau pour une moyenne de fréquence de type de mutation par génération (à partir de la fréquence des échantillons) : df_freq_moy_gen
même chose mais par traitement, par génération: df_freq_moy_gen_traitement : df_freq_moy_gen_traitement


nb_indel_ech_gen_t -> fréquence relative moyenne des indels dans chaque traitement de chaque génération
nb_indel_ech_gen -> fréquence relative moyenne des indels dans chaque génération




ICI

choisir les graphes : 
frequence moyenne des indels par éch au cours des générations et au au cours des générations / par traitement
Fréquence moyenne des indels par génération (les barplot en dessous) 
distribution nb des indels en boxplot (en dessous -> choisir)
Regarder si la longeur moyenne des indels tend à augmenter au cours du temps ET par traitement (deux graphes)


```{r}
ggplot(data = count_ins_del_sv, aes(x = GENERATION, y = COUNT_TYPE, color=TYPE)) +
  geom_boxplot(position=position_dodge(1), outlier.colour="black", outlier.shape=16,
             outlier.size=2, outlier.alpha = 0.5 ,notch=FALSE) +
  geom_dotplot(aes(fill = TYPE), binaxis='y', stackdir='center', position=position_dodge(), alpha = 0.5, dotsize = 0.5, show.legend = FALSE) +
  scale_color_brewer(palette="Dark2") +
  scale_fill_brewer(palette="Dark2") + 
  labs(x = "Génération", y = "Effectif des types de mutations (délétions ou insertions)", color = "DEL = délétion /\n INS = insértion") +
  ggtitle("Distribution des boîtes à moustaches des délétions et\n des insértions de variants structuraux  par génération") +
  theme_bw()+
  theme(plot.title = element_text(hjust = 0.5))


ggplot(data = count_ins_del_sv, aes(x = GENERATION, y = COUNT_TYPE, color=TRAITEMENT)) +
  geom_boxplot(position=position_dodge(1), outlier.colour="black", outlier.shape=16,
             outlier.size=2, outlier.alpha = 0.5 ,notch=FALSE) +
  geom_dotplot(aes(fill = TYPE), binaxis='y', stackdir='center', position=position_dodge(), alpha = 0.7, dotsize = 0.5, show.legend = TRUE) +
  scale_color_brewer(palette="Dark2") +
  scale_fill_brewer(palette="Dark2") + 
  labs(x = "Génération", y = "Effectif des types de mutations\n(délétions ou insertions)", color = "Type de traitement") +
  ggtitle("Distribution des boîtes à moustaches des délétions et des\ninsértions de variants structuraux  par type de traitement et par génération") +
  facet_wrap(~TYPE) +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5))
```


## Les SV
compter le nb d'insertion et le nombre de deletion par échantillon et génération
on peut faire le % d'insertion aussi
faire leur longueur moyenne, medianne avec les max, les min les sq
rajouter traitement chaud, traitement froid et pas de traitement



Regarder la distribution en termes de % d’effectifs de mutations

Regarder la longueur des mutations en fonction de si c'est del ou ins et en fonction du traitement et des génération
```{r}
table_sv %>%
  ggplot(aes(x = GENERATION, y = abs(LONGUEUR), by = TRAITEMENT, color = TRAITEMENT))+
  geom_boxplot(position = position_dodge(1))+
  #geom_dotplot(aes(fill = TRAITEMENT), binaxis='y', stackdir='center', position=position_dodge(1), alpha = 0.8, dotsize = 0.7) + 
  scale_color_brewer(palette="Dark2") +
  scale_fill_brewer(palette="Dark2") +
  scale_y_log10() +
  theme_bw()

table_sv %>%
  ggplot(aes(x = GENERATION, y = abs(LONGUEUR), by = TRAITEMENT, color = TRAITEMENT))+
  geom_boxplot(position = position_dodge(1))+
  #geom_dotplot(aes(fill = TRAITEMENT), binaxis='y', stackdir='center', position=position_dodge(1), alpha = 0.8, dotsize = 0.7) + 
  scale_color_brewer(palette="Dark2") +
  scale_fill_brewer(palette="Dark2") +
  facet_wrap(~TYPE.DE.MUTATION)+
  scale_y_log10() +
  theme_bw()
```


Je veux représenter la distribution des longueurs de mutations en fonction de leur type, de la génération et du traitement

```{r}
table_sv %>% names()
table_sv <- table_sv %>% mutate(LONGUEUR_abs = round(abs(LONGUEUR),3))

df_length_mut_sv <- table_sv %>% aggregate(LONGUEUR_abs ~ GENERATION + ECHANTILLON +TRAITEMENT  + TYPE.DE.MUTATION , FUN = mean)
df2 <- table_sv %>% aggregate(LONGUEUR_abs ~ GENERATION + ECHANTILLON + TRAITEMENT + TYPE.DE.MUTATION , FUN = median)
df3 <- table_sv %>% aggregate(LONGUEUR_abs ~ GENERATION + ECHANTILLON + TRAITEMENT + TYPE.DE.MUTATION , FUN = sd)

names(df_length_mut_sv)[names(df_length_mut_sv) == "LONGUEUR_abs"] <- "LONGUEUR_abs_mean"
names(df2)[names(df2) == "LONGUEUR_abs"] <- "LONGUEUR_abs_median"
names(df3)[names(df3) == "LONGUEUR_abs"] <- "LONGUEUR_abs_sd"

df_length_mut_sv <- left_join(df_length_mut_sv, df2)
df_length_mut_sv <- left_join(df_length_mut_sv, df3)  
remove(df2, df3)

df_length_mut_sv <- df_length_mut_sv %>% mutate(LONGUEUR_abs_mean = round(LONGUEUR_abs_mean,3))
df_length_mut_sv <- df_length_mut_sv %>% mutate(LONGUEUR_abs_sd = round(LONGUEUR_abs_sd,3))
```




```{r}
table_sv$LONGUEUR_abs %>% summary()
table_sv$LONGUEUR_abs %>% length() #698
table_sv[which(table_sv$LONGUEUR_abs <= 100) , ] %>% nrow #662
table_sv[which(table_sv$LONGUEUR_abs > 100) , ] %>% nrow() #36
table_sv[which(table_sv$LONGUEUR_abs > 100) , "LONGUEUR_abs"] %>% sort()

#Distribution de la longueur des mutations
table_sv[which(table_sv$LONGUEUR_abs <= 250) , ] %>% 
  ggplot(aes(x = LONGUEUR_abs)) +
  geom_histogram(binwidth=1, fill = "aquamarine3", color = "aquamarine4") +
  geom_vline(aes(xintercept=mean(LONGUEUR_abs)), color="blue", linetype="dashed", size=1) +
  #geom_text(aes(label = LONGUEUR_abs), stat = "count", vjust = -0.7 , size = 2, color = "black") +
  #scale_y_log10() +
  #scale_x_log10() +
  theme_bw()

# J'ai colorié en fonction de si c'est une délétion ou un insertion
table_sv[which(table_sv$LONGUEUR_abs <= 250) , ] %>% 
  ggplot(aes(x = LONGUEUR_abs, color = TYPE.DE.MUTATION, fill = TYPE.DE.MUTATION)) + #
  geom_histogram(binwidth=1, alpha=0.5, position="dodge") +
  scale_color_brewer(palette="Accent") + 
   scale_fill_brewer(palette="Accent") + 
  #geom_vline(aes(xintercept=mean(LONGUEUR_abs)), color="blue", linetype="dashed", size=1) +
  #scale_color_manual(values= wes_palette("Royal2", n = 2)) +
  theme_bw()

# toutes les mutations confondues mais on prend en compte la génération
table_sv[which(table_sv$LONGUEUR_abs <= 250) , ] %>%
  ggplot(aes(x = LONGUEUR_abs)) +
  geom_histogram(binwidth=1, fill = "aquamarine3", color = "aquamarine4") +
  facet_wrap(~GENERATION, scales = "free") +
  theme_bw()
```


```{r}
#Là c'est la distribution des longueurs de mutations en fonction des générations et j'ai colorié les insertions et les délétions
table_sv[which(table_sv$LONGUEUR_abs <= 250) , ] %>% 
  ggplot(aes(x = LONGUEUR_abs, color = TYPE.DE.MUTATION, fill = TYPE.DE.MUTATION)) + #
  geom_histogram(binwidth=1, alpha=0.5, position=position_dodge()) +
  scale_color_brewer(palette="Accent") + 
  scale_fill_brewer(palette="Accent") + 
  facet_wrap(~GENERATION, scales = "free") +
  ggtitle("longueurs de mutations inférieures à 250 pb") +
  theme_bw()

table_sv[which(table_sv$LONGUEUR_abs >= 250) , ] %>% 
  ggplot(aes(x = LONGUEUR_abs, color = TYPE.DE.MUTATION, fill = TYPE.DE.MUTATION)) + #
  geom_histogram(binwidth=1, alpha=0.5, position=position_dodge()) +
  scale_color_brewer(palette="Accent") + 
  scale_fill_brewer(palette="Accent") + 
  facet_wrap(~GENERATION, scales = "free") +
  ggtitle("longueurs de mutations supérieures à 250 pb") +
  theme_bw()

# les distributions de longueurs de mutations en fonction des traitements
table_sv[which(table_sv$LONGUEUR_abs <= 250) , ] %>% 
  ggplot(aes(x = LONGUEUR_abs, color =TRAITEMENT , fill = TRAITEMENT)) + #
  geom_histogram(binwidth=1, alpha=0.5, position=position_dodge()) +
  scale_color_brewer(palette="Accent") + 
  scale_fill_brewer(palette="Accent") + 
  facet_wrap(~TYPE.DE.MUTATION, scales = "free") +
  theme_bw()
```


```{r}
#distribution des longueurs de mutation en fonction de leur type (insertion ou délétion)et par traitement
table_sv[which(table_sv$LONGUEUR_abs <= 250 & table_sv$GENERATION == "P15") , ] %>% 
  ggplot(aes(x = LONGUEUR_abs, color =TRAITEMENT , fill = TRAITEMENT)) + #
  geom_histogram(binwidth=2, alpha=0.5, position="dodge") +
  #geom_vline(data=df_length_mut_sv[which(df_length_mut_sv$GENERATION == "P15"),], aes(xintercept=LONGUEUR_abs_mean, color= TRAITEMENT), linetype="dashed", size = 1)+
  scale_color_brewer(palette="Accent") + 
  scale_fill_brewer(palette="Accent") + 
  facet_wrap(~TYPE.DE.MUTATION) +
  ggtitle("P15") + 
  #facet_wrap(~traitement_type, scales = "free") +
  theme_bw()


table_sv[which(table_sv$LONGUEUR_abs <= 250 & table_sv$GENERATION == "P30") , ] %>% 
  ggplot(aes(x = LONGUEUR_abs, color =TRAITEMENT , fill = TRAITEMENT)) + #
  geom_histogram(binwidth=2, alpha=0.5, position="dodge") +
 # geom_vline(data=df_length_mut_sv[which(df_length_mut_sv$GENERATION == "P30"),], aes(xintercept=LONGUEUR_abs_mean, color= TRAITEMENT), linetype="dashed", size = 1)+
  scale_color_brewer(palette="Accent") + 
  scale_fill_brewer(palette="Accent") + 
  ggtitle("P30") + 
  facet_wrap(~TYPE.DE.MUTATION, scales = "free") +
  theme_bw()


table_sv[which(table_sv$LONGUEUR_abs <= 250 & table_sv$GENERATION == "P50") , ] %>% 
  ggplot(aes(x = LONGUEUR_abs, color =TRAITEMENT , fill = TRAITEMENT)) + #
  geom_histogram(binwidth=2, alpha=0.5, position="dodge") +
 # geom_vline(data=df_length_mut_sv[which(df_length_mut_sv$GENERATION == "P50"),], aes(xintercept=LONGUEUR_abs_mean, color= TRAITEMENT), linetype="dashed", size = 1)+
  scale_color_brewer(palette="Accent") + 
  scale_fill_brewer(palette="Accent") + 
  ggtitle("P50") + 
  facet_wrap(~TYPE.DE.MUTATION, scales = "free") +
  theme_bw()

table_sv[which(table_sv$LONGUEUR_abs <= 250 & table_sv$GENERATION == "P65") , ] %>% 
  ggplot(aes(x = LONGUEUR_abs, color =TRAITEMENT , fill = TRAITEMENT)) + #
  geom_histogram(binwidth=2, alpha=0.5, position="dodge") +
  #geom_vline(data=df_length_mut_sv[which(df_length_mut_sv$GENERATION == "P65"),], aes(xintercept=LONGUEUR_abs_mean, color= TRAITEMENT), linetype="dashed", size = 1)+
  scale_color_brewer(palette="Accent") + 
  scale_fill_brewer(palette="Accent") + 
  ggtitle("P65") + 
  facet_wrap(~TYPE.DE.MUTATION, scales = "free") +
  theme_bw()

table_sv[which(table_sv$LONGUEUR_abs <= 250 & table_sv$GENERATION == "P90") , ] %>% 
  ggplot(aes(x = LONGUEUR_abs, color =TRAITEMENT , fill = TRAITEMENT)) + #
  geom_histogram(binwidth=2, alpha=0.5, position="dodge") +
 # geom_vline(data=df_length_mut_sv[which(df_length_mut_sv$GENERATION == "P90"),], aes(xintercept=LONGUEUR_abs_mean, color= TRAITEMENT), linetype="dashed", size = 1)+
  scale_color_brewer(palette="Accent") + 
  scale_fill_brewer(palette="Accent") + 
  ggtitle("P90") + 
  facet_wrap(~TYPE.DE.MUTATION, scales = "free") +
  theme_bw()

```


```{r}
df_length_mut_sv %>% 
  ggplot(aes(x = GENERATION, y = LONGUEUR_abs_mean, color = TRAITEMENT))+
  geom_boxplot()+
  scale_color_brewer(palette = "Dark2", name = "Traitement", labels = c("Aucun", "Chaud", "Froid"))+
  facet_wrap(~TYPE.DE.MUTATION) +
  scale_y_log10()+
  labs(y = "longueur\nabsolue\nmoyenne", x = "Génération", title = "Statistiques des longueurs absolues moyennes des\ninsertions et des délétions par génération") +
  theme_bw() +
  theme(axis.title.y = element_text(angle = 0, vjust = 0.5))

table_sv %>%
  ggplot(aes(x = GENERATION, y = abs(LONGUEUR), by = TRAITEMENT, color = TRAITEMENT))+
  geom_boxplot(position = position_dodge(1))+
  #geom_dotplot(aes(fill = TRAITEMENT), binaxis='y', stackdir='center', position=position_dodge(1), alpha = 0.8, dotsize = 0.7) + 
  scale_color_brewer(palette="Dark2", name = "Traitement", labels = c("Aucun", "Chaud", "Froid")) +
  #scale_fill_brewer(palette="Dark2") +
  labs(y = "longueur\nabsolue\nmoyenne", x = "Génération", title = "Statistiques des longueurs absolues moyennes des\ninsertions et des délétions par génération et par traitement") +
  facet_wrap(~TYPE.DE.MUTATION)+
  scale_y_log10() +
  theme_bw() +
  
  theme(axis.title.y = element_text(angle = 0, vjust = 0.5))

#ggsave(".\\images_scripts\\boxplot_longueur_mean_ins_del.png", height = 4, width = 7)
  
```


```{r}
## moyenne de moyenne des longueurs par , generation
df_length_moy_g <- df_length_mut_sv %>% aggregate(LONGUEUR_abs_mean ~ GENERATION , FUN = mean)
df_length_moy_g <- df_length_moy_g %>% mutate(GEN_num = case_when(df_length_moy_g$GENERATION == "P15"~15,
                                                                        df_length_moy_g$GENERATION == "P30"~30,
                                                                        df_length_moy_g$GENERATION == "P50"~50,
                                                                        df_length_moy_g$GENERATION == "P65"~65,
                                                                        df_length_moy_g$GENERATION == "P90"~90,
                                                                        ))
df_length_moy_g$LONGUEUR_abs_mean <- round(df_length_moy_g$LONGUEUR_abs_mean,3)

df_length_moy_g %>%
  ggplot(aes(x = GEN_num, y = LONGUEUR_abs_mean)) +
  geom_line()+
  geom_point()+
  #facet_wrap(~TYPE.DE.MUTATION)+
  geom_text(aes(label = LONGUEUR_abs_mean), stat = "identity", vjust = -1 , size = 2, color = "black")+
  scale_color_brewer(palette = "Dark2")+
  scale_y_log10()+
  theme_bw()



## moyenne de moyenne des longueurs par , generation et type de mutation
df_length_moy_mug <- df_length_mut_sv %>% aggregate(LONGUEUR_abs_mean ~ GENERATION + TYPE.DE.MUTATION, FUN = mean)
df_length_moy_mug <- df_length_moy_mug %>% mutate(GEN_num = case_when(df_length_moy_mug$GENERATION == "P15"~15,
                                                                        df_length_moy_mug$GENERATION == "P30"~30,
                                                                        df_length_moy_mug$GENERATION == "P50"~50,
                                                                        df_length_moy_mug$GENERATION == "P65"~65,
                                                                        df_length_moy_mug$GENERATION == "P90"~90,
                                                                        ))
df_length_moy_mug$LONGUEUR_abs_mean <- round(df_length_moy_mug$LONGUEUR_abs_mean,3)

df_length_moy_mug %>%
  ggplot(aes(x = GEN_num, y = LONGUEUR_abs_mean, color = TYPE.DE.MUTATION)) +
  geom_line()+
  geom_point()+
  #facet_wrap(~TYPE.DE.MUTATION)+
  geom_text(aes(label = LONGUEUR_abs_mean), stat = "identity", vjust = -1 , size = 2, color = "black")+
  scale_color_brewer(palette = "Dark2")+
  scale_y_log10()+
  theme_bw()

## moyenne de moyenne des longueurs par traitement, generation
df_length_moy_tg <- df_length_mut_sv %>% aggregate(LONGUEUR_abs_mean ~ GENERATION + TRAITEMENT , FUN = mean)
df_length_moy_tg <- df_length_moy_tg %>% mutate(GEN_num = case_when(df_length_moy_tg$GENERATION == "P15"~15,
                                                                        df_length_moy_tg$GENERATION == "P30"~30,
                                                                        df_length_moy_tg$GENERATION == "P50"~50,
                                                                        df_length_moy_tg$GENERATION == "P65"~65,
                                                                        df_length_moy_tg$GENERATION == "P90"~90,
                                                                        ))
df_length_moy_tg$LONGUEUR_abs_mean <- round(df_length_moy_tg$LONGUEUR_abs_mean,3)

df_length_moy_tg %>%
  ggplot(aes(x = GEN_num, y = LONGUEUR_abs_mean, color = TRAITEMENT)) +
  geom_line()+
  geom_point()+
  #facet_wrap(~TYPE.DE.MUTATION)+
  geom_text(aes(label = LONGUEUR_abs_mean), stat = "identity", vjust = -1 , size = 2, color = "black")+
  scale_color_brewer(palette = "Dark2")+
  scale_y_log10()+
  theme_bw()



## moyenne de moyenne des longueurs par type d emutation, traitement, generation
df_length_moy_mutg <- df_length_mut_sv %>% aggregate(LONGUEUR_abs_mean ~ GENERATION + TRAITEMENT + TYPE.DE.MUTATION, FUN = mean)
df_length_moy_mutg <- df_length_moy_mutg %>% mutate(GEN_num = case_when(df_length_moy_mutg$GENERATION == "P15"~15,
                                                                        df_length_moy_mutg$GENERATION == "P30"~30,
                                                                        df_length_moy_mutg$GENERATION == "P50"~50,
                                                                        df_length_moy_mutg$GENERATION == "P65"~65,
                                                                        df_length_moy_mutg$GENERATION == "P90"~90,
                                                                        ))
df_length_moy_mutg$LONGUEUR_abs_mean <- round(df_length_moy_mutg$LONGUEUR_abs_mean,3)
```


```{r}
df_length_moy_mutg %>%
  ggplot(aes(x = GEN_num, y = LONGUEUR_abs_mean, color = TRAITEMENT)) +
  geom_line()+
  geom_point()+
  facet_wrap(~TYPE.DE.MUTATION)+
  geom_text(aes(label = LONGUEUR_abs_mean), stat = "identity", vjust = -1 , size = 2, color = "black")+
  scale_color_brewer(palette = "Dark2", name = "Traitement", labels = c("Aucun", "Chaud", "Froid"))+
  labs(y = "longueur\nabsolue\nmoyenne", x = "Génération", title = "Statistiques de moyennes de longueurs absolues des\ninsertions et des délétions par génération et par traitement", subtitle = "(moyenne de moyenne sur les échantillons)") +
  scale_y_log10()+
  theme_bw() + 
  theme(axis.title.y = element_text(angle = 0, vjust = 0.5))

#ggsave(".\\images_scripts\\boxplot_longueur_mean_mean_ins_del.png", height = 4, width = 7)
```


```{r}
remove(df_length_moy_g, df_length_moy_mug, df_length_moy_mutg, df_length_moy_tg)
```

Il y a des générations où les longueurs de mutations sont de plus de 1000 paires de bases 
Je vais faire deux sous tableaux : un qui aura la moyenne des longueurs de sv <250 pb et un autre qui aura la moyenne des longueurs de sv > 250 (en fait je vais pas faire le deuxième parce que c'est pas utile, il y en a 12 en tout...)
`

Les distributions des longueurs de mutations sont super proches quand on zoom sur des longueurs inférieures à 250 pb, les moyennes sont assez proches (en sachant que là c'est la distribution des longueurs par génération tout échantillon confondus...)
Si je regarde la distribution des moyennes des échantillons : 







